From f2d50a9d5bfa0f2e657f820e6c3b8cc5fe4e42cb Mon Sep 17 00:00:00 2001
From: WangJing <wangjing@mprc.pku.edu.cn>
Date: Tue, 25 Oct 2011 18:00:06 +0800
Subject: [PATCH 026/641] UniCore64: Modify unicore64/Makefile to compile mm

Signed-off-by: WangJing <wangjing@mprc.pku.edu.cn>
---
 arch/unicore64/Makefile |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/unicore64/Makefile b/arch/unicore64/Makefile
index b3e5b3f..4f9bbce 100644
--- a/arch/unicore64/Makefile
+++ b/arch/unicore64/Makefile
@@ -30,7 +30,7 @@ KBUILD_CFLAGS		+= -msoft-float
 #head-y			+= arch/unicore64/kernel/init_task.o
 
 core-y			+= arch/unicore64/kernel/
-#core-y			+= arch/unicore64/mm/
+core-y			+= arch/unicore64/mm/
 
 #libs-y			+= arch/unicore64/lib/
 
-- 
1.7.9.5

