From 1cccf0957994aefbfadc48b04c56a7e5fc02877b Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Fri, 25 Nov 2011 14:57:11 +0800
Subject: [PATCH 088/641] UniCore64: add HEX64 macro for 64-bit address
 definition

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/include/arch/bitfield.h |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/unicore64/include/arch/bitfield.h b/arch/unicore64/include/arch/bitfield.h
index e598aea..62e8d6b 100644
--- a/arch/unicore64/include/arch/bitfield.h
+++ b/arch/unicore64/include/arch/bitfield.h
@@ -13,4 +13,10 @@
 #define FMASK(vmask, vshift)			\
 		(((_AC(1, UL) << (vmask)) - 1) << (vshift))
 
+/*
+ * To define 64-bit addresses:
+ *   HEX64(1f, ffffffff) means 0x1fffffffff
+ */
+#define HEX64(hi, lo)		_AC(0x ## hi ## lo, UL)
+
 #endif /* __UNICORE64_ARCH_BITFIELD_H__ */
-- 
1.7.9.5

