From d4b598085c51cb2ae17ab420a3e94af2b2dea2be Mon Sep 17 00:00:00 2001
From: Yu Yue <yuyue@mprc.pku.edu.cn>
Date: Wed, 30 Nov 2011 11:35:43 +0800
Subject: [PATCH 100/641] UniCore64: remove nop8 defination in asm-common.h

Signed-off-by: Yu Yue <yuyue@mprc.pku.edu.cn>
---
 arch/unicore64/include/arch/asm-common.h |   10 ----------
 1 file changed, 10 deletions(-)

diff --git a/arch/unicore64/include/arch/asm-common.h b/arch/unicore64/include/arch/asm-common.h
index 898a3ef..23492bc 100644
--- a/arch/unicore64/include/arch/asm-common.h
+++ b/arch/unicore64/include/arch/asm-common.h
@@ -1,16 +1,6 @@
 #ifndef __UNICORE64_ARCH_ASM_COMMON_H__
 #define __UNICORE64_ARCH_ASM_COMMON_H__
 
-#ifndef __ASSEMBLY__
-#error "Only include this from assembly code"
-#endif /* __ASSEMBLY__ */
-
 #define csub		cmpsub
 
-	.macro	nop8
-	.rept	8
-		nop
-	.endr
-	.endm
-
 #endif /* __UNICORE64_ARCH_ASM_COMMON_H__ */
-- 
1.7.9.5

