From 4237f207bf58c032b9eb0ca8803cda125dabb3c9 Mon Sep 17 00:00:00 2001
From: Yu Yue <yuyue@mprc.pku.edu.cn>
Date: Wed, 30 Nov 2011 11:35:44 +0800
Subject: [PATCH 101/641] UniCore64: add __ASM_WRAP

Signed-off-by: Yu Yue <yuyue@mprc.pku.edu.cn>
---
 arch/unicore64/include/arch/asm-common.h |   10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/unicore64/include/arch/asm-common.h b/arch/unicore64/include/arch/asm-common.h
index 23492bc..849e6fb 100644
--- a/arch/unicore64/include/arch/asm-common.h
+++ b/arch/unicore64/include/arch/asm-common.h
@@ -3,4 +3,14 @@
 
 #define csub		cmpsub
 
+/*
+ * __ASM_WRAP is ONLY used for assembly macro definitions.
+ * Then this macro could be used in both assembly files and c files.
+ */
+#ifdef __ASSEMBLY__
+#define __ASM_WRAP(code...)	code
+#else
+#define __ASM_WRAP(code...)	__asm__(#code);
+#endif /* __ASSEMBLY__ */
+
 #endif /* __UNICORE64_ARCH_ASM_COMMON_H__ */
-- 
1.7.9.5

