From 2afa8f8d91b55c13bf9da2524497c29e0f5660df Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Tue, 6 Dec 2011 10:47:05 +0800
Subject: [PATCH 119/641] UniCore64: modify archhelp typo

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/Makefile |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/unicore64/Makefile b/arch/unicore64/Makefile
index c3e0532..ca18d1d 100644
--- a/arch/unicore64/Makefile
+++ b/arch/unicore64/Makefile
@@ -41,6 +41,6 @@ archdoc:
 	$(Q)$(MAKE) $(build)=$(DOC_DIR) DOCBOOKS=unicore64-kernel.xml pdfdocs
 
 define archhelp
-	echo  '  defconfig     - using arch/$(ARCH)/configs/$(KBUILD_DEFCONFIG)'
+	echo  '  defconfig     - Use arch/$(ARCH)/configs/$(KBUILD_DEFCONFIG)'
 	echo  '  archdoc       - Generate $(DOC_DIR)/unicore64-kernel.pdf'
 endef
-- 
1.7.9.5

