From b989d1ce98c7148e75a86a526dc3044e8f8301bb Mon Sep 17 00:00:00 2001
From: Liu Guoli <liuguoli@mprc.pku.edu.cn>
Date: Wed, 7 Dec 2011 11:53:12 +0800
Subject: [PATCH 125/641] UniCore64: Add kernel-doc for irqflags.h

Signed-off-by: Liu Guoli <liuguoli@mprc.pku.edu.cn>
---
 arch/unicore64/include/asm/irqflags.h |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/arch/unicore64/include/asm/irqflags.h b/arch/unicore64/include/asm/irqflags.h
index 84835ab..229c708 100644
--- a/arch/unicore64/include/asm/irqflags.h
+++ b/arch/unicore64/include/asm/irqflags.h
@@ -8,6 +8,9 @@
 
 /**
  * arch_local_save_flags() - Save the current interrupt enable state.
+ *
+ * Return codes:
+ * Current CPU interrupt state.
  */
 static inline unsigned long arch_local_save_flags(void)
 {
@@ -20,7 +23,7 @@ static inline unsigned long arch_local_save_flags(void)
 
 /**
  * arch_local_irq_restore() - restore saved IRQ state
- * @flags:
+ * @flags: CPU interrupt mask
  */
 static inline void arch_local_irq_restore(unsigned long flags)
 {
-- 
1.7.9.5

