From 1029c8046dcd5ec1bef8608320bd79b5b692d38f Mon Sep 17 00:00:00 2001
From: LiangRongrong <Liangrongrong@mprc.pku.edu.cn>
Date: Fri, 9 Dec 2011 18:11:35 +0800
Subject: [PATCH 132/641] UniCore64: Add PGD address definition

Signed-off-by: Liang Rongrong <liangrongrong@mprc.pku.edu.cn>
---
 arch/unicore64/include/arch/hwdef-memory.h |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/unicore64/include/arch/hwdef-memory.h b/arch/unicore64/include/arch/hwdef-memory.h
index b6002ed..f880578 100644
--- a/arch/unicore64/include/arch/hwdef-memory.h
+++ b/arch/unicore64/include/arch/hwdef-memory.h
@@ -61,5 +61,11 @@
 
 /* The byte offset of the kernel image in RAM from the start of RAM. */
 #define UC64_PM_KIMAGE_START		HEX64(00000000, 00408000)
+#define UC64_VM_KIMAGE_START		(UC64_VM_KERNEL_START + \
+						UC64_PM_KIMAGE_START)
+
+/* physical and virtual address of the initial page table */
+#define UC64_PM_PGTABLE_PGD		(UC64_PM_KIMAGE_START - 0x1000)
+#define UC64_VM_PGTABLE_PGD		(UC64_VM_KIMAGE_START - 0x1000)
 
 #endif /* __UNICORE64_ARCH_HWDEF_MEMORY_H__ */
-- 
1.7.9.5

