From 503dea5c0213c456096bf00bab4c674a7ca41c96 Mon Sep 17 00:00:00 2001
From: Liu Guoli <liuguoli@mprc.pku.edu.cn>
Date: Fri, 9 Dec 2011 17:47:56 +0800
Subject: [PATCH 134/641] UniCore64: Add block doc for irqflags.h

Signed-off-by: Liu Guoli <liuguoli@mprc.pku.edu.cn>
---
 arch/unicore64/include/asm/irqflags.h |   18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/arch/unicore64/include/asm/irqflags.h b/arch/unicore64/include/asm/irqflags.h
index 229c708..71e9643 100644
--- a/arch/unicore64/include/asm/irqflags.h
+++ b/arch/unicore64/include/asm/irqflags.h
@@ -1,6 +1,24 @@
 #ifndef __UNICORE64_ASM_IRQFLAGS_H__
 #define __UNICORE64_ASM_IRQFLAGS_H__
 
+/**
+ * DOC: IRQFLAGS_H
+ * Code for interrupt control specific to PKUnity SoC and UniCore ISA.
+ * Use the generic function for simple life.
+ *
+ * \\\\lt:programlisting\\\\gt:
+ * -
+ * arch_local_save_flags():		read interrupt enabled status
+ * arch_local_irq_restore():	set interrupt enabled status
+ * arch_local_irq_save():		get status and disable interrupts
+ * arch_irqs_disabled_flags():	test flags
+ * arch_local_irq_enable():		unconditionally enable interrupts
+ * arch_local_irq_disable():	unconditionally disable interrupts
+ * arch_irqs_disabled():		test hardware interrupt enable bit
+ * -
+ * \\\\lt:/programlisting\\\\gt:
+ */
+
 #include <arch/hwdef-cpu.h>
 
 #define ARCH_IRQ_DISABLED	ASR_INTR_MASK
-- 
1.7.9.5

