From 505e9a32788914dac7b176342e3097c37c622da3 Mon Sep 17 00:00:00 2001
From: Liu Guoli <liuguoli@mprc.pku.edu.cn>
Date: Fri, 9 Dec 2011 17:47:58 +0800
Subject: [PATCH 135/641] UniCore64: Add kernel-doc for vmlinux.lds.S

Signed-off-by: Liu Guoli <liuguoli@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/vmlinux.lds.S |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/unicore64/kernel/vmlinux.lds.S b/arch/unicore64/kernel/vmlinux.lds.S
index cd95dec..39f39e8 100644
--- a/arch/unicore64/kernel/vmlinux.lds.S
+++ b/arch/unicore64/kernel/vmlinux.lds.S
@@ -1,3 +1,8 @@
+/**
+ * DOC: VMLINUX_LDS_S
+ * ld script to make UniCore64 Linux kernel
+ */
+
 #include <asm-generic/vmlinux.lds.h>
 #include <asm/thread_info.h>
 #include <asm/page.h>
-- 
1.7.9.5

