From bf1e4da8781ed98e8d54327ce3ff66c2dde04193 Mon Sep 17 00:00:00 2001
From: Liu Guoli <liuguoli@mprc.pku.edu.cn>
Date: Fri, 9 Dec 2011 17:47:57 +0800
Subject: [PATCH 136/641] UniCore64: Delete __attribute__((aligned(8)))

Signed-off-by: Liu Guoli <liuguoli@mprc.pku.edu.cn>
---
 arch/unicore64/include/asm/elf.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/unicore64/include/asm/elf.h b/arch/unicore64/include/asm/elf.h
index 99922ec..97c06e4 100644
--- a/arch/unicore64/include/asm/elf.h
+++ b/arch/unicore64/include/asm/elf.h
@@ -21,7 +21,7 @@ typedef elf_greg_t elf_gregset_t[ELF_NGREG];
 
 struct fpu_regs {
 	unsigned int regs[FP_REGS_NUMBER];
-} __attribute__((aligned(8)));
+};
 
 #endif
 
-- 
1.7.9.5

