From fe6496c135911c965010585f045ccd51b298ae11 Mon Sep 17 00:00:00 2001
From: GuanXuetao <gxt@mprc.pku.edu.cn>
Date: Sat, 10 Dec 2011 13:33:09 +0800
Subject: [PATCH 140/641] UniCore64: correct PAGE_MASK misunderstanding

Signed-off-by: GuanXuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/include/asm/page.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/unicore64/include/asm/page.h b/arch/unicore64/include/asm/page.h
index 45f7404..84bc093 100644
--- a/arch/unicore64/include/asm/page.h
+++ b/arch/unicore64/include/asm/page.h
@@ -10,7 +10,7 @@
 /* PAGE_SHIFT determines the page size */
 #define PAGE_SHIFT		UC64_PAGE_BITS
 #define PAGE_SIZE		UC64_PAGE_SIZE
-#define PAGE_MASK		UC64_PAGE_MASK
+#define PAGE_MASK		(~(UC64_PAGE_MASK))
 
 #ifndef __ASSEMBLY__
 
-- 
1.7.9.5

