From 78052ea16d01131c64bdaccfe3341f6c31a31c96 Mon Sep 17 00:00:00 2001
From: LiangRongrong <Liangrongrong@mprc.pku.edu.cn>
Date: Thu, 15 Dec 2011 15:48:12 +0800
Subject: [PATCH 174/641] UniCore64: Add hwdef-pgtable.h kernel-doc support

Signed-off-by: Liang Rongrong <liangrongrong@mprc.pku.edu.cn>
---
 Documentation/DocBook/unicore64-kernel.tmpl |   11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/Documentation/DocBook/unicore64-kernel.tmpl b/Documentation/DocBook/unicore64-kernel.tmpl
index 9d6e5a9..e77b30e 100644
--- a/Documentation/DocBook/unicore64-kernel.tmpl
+++ b/Documentation/DocBook/unicore64-kernel.tmpl
@@ -73,6 +73,17 @@
 !Parch/unicore64/include/arch/hwdef-memory.h HWDEF_MEMORY_H_VM
 !Parch/unicore64/include/arch/hwdef-memory.h HWDEF_MEMORY_H_PM
 		</sect2>
+		<sect2 id="sect1_1_4">
+			<title>Page table</title>
+			<screen>
+			See file: arch/unicore64/include/arch/hwdef-pgtable.h
+			</screen>
+!Parch/unicore64/include/arch/hwdef-pgtable.h HWDEF_PGTABLE_H
+!Parch/unicore64/include/arch/hwdef-pgtable.h HWDEF_PGTABLE_H_ADDR_TRANSLATION
+!Parch/unicore64/include/arch/hwdef-pgtable.h HWDEF_PGTABLE_H_PGD
+!Parch/unicore64/include/arch/hwdef-pgtable.h HWDEF_PGTABLE_H_PMD
+!Parch/unicore64/include/arch/hwdef-pgtable.h HWDEF_PGTABLE_H_PTE
+		</sect2>
     	</sect1>
 </chapter>
 
-- 
1.7.9.5

