From c8220c84d7756d606755b5b329d55e149954dbd3 Mon Sep 17 00:00:00 2001
From: GuanXuetao <gxt@mprc.pku.edu.cn>
Date: Fri, 16 Dec 2011 20:11:00 +0800
Subject: [PATCH 179/641] UniCore64: adjust the format of boot.lds.S

Signed-off-by: GuanXuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/boot/boot.lds.S |   47 +++++++++++++++++++---------------------
 1 file changed, 22 insertions(+), 25 deletions(-)

diff --git a/arch/unicore64/boot/boot.lds.S b/arch/unicore64/boot/boot.lds.S
index 0547397..2c42d4b 100644
--- a/arch/unicore64/boot/boot.lds.S
+++ b/arch/unicore64/boot/boot.lds.S
@@ -4,38 +4,35 @@ OUTPUT_ARCH(unicore64)
 ENTRY(_start)
 SECTIONS
 {
-  /DISCARD/ : {
-    /*
-     * Discard any r/w data - this produces a link error if we have any,
-     * which is required for PIC decompression.  Local data generates
-     * GOTOFF relocations, which prevents it being relocated independently
-     * of the text/got segments.
-     */
-    *(.data)
-  }
+	/DISCARD/ : {
+		/*
+		 * Discard any r/w data -
+		 * This produces a link error if we have any, which is
+		 * required for PIC decompression.  Local data generates
+		 * GOTOFF relocations, which prevents it being relocated
+		 * independently of the text/got segments.
+		 */
+		*(.data)
+	}
 
-  . = UC64_PM_ZIMAGE_START;
-  _text = .;
+	. = UC64_PM_ZIMAGE_START;
 
-  .text : {
-    _start = .;
-    *(.start)
-    *(.text)
-    *(.rodata)
-    *(.rodata.*)
+	.text : {
+		_start = .;
+		*(.start)
+		*(.text)
+		*(.rodata)
+		*(.rodata.*)
 
 		input_data = .;
 		*(.piggydata)
 		input_data_end = .;
 
-    . = ALIGN(4);
-  }
-  _etext = .;
+		. = ALIGN(4);
+	}
 
-  . = ALIGN(4);
-  __bss_start = .;
-  .bss			: { *(.bss) }
-  _end = .;
+	__bss_start = .;
+	.bss : { *(.bss) }
+	_end = .;
 
-  .comment 0		: { *(.comment) }
 }
-- 
1.7.9.5

