From c0323f17826000c15002aa41735ab02c5d4a5d6b Mon Sep 17 00:00:00 2001
From: Liu Guoli <liuguoli@mprc.pku.edu.cn>
Date: Mon, 19 Dec 2011 11:37:56 +0800
Subject: [PATCH 183/641] UniCore64: Add kernel-doc support for vmlinux.lds.S

Signed-off-by: Liu Guoli <liuguoli@mprc.pku.edu.cn>
---
 Documentation/DocBook/unicore64-kernel.tmpl |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/Documentation/DocBook/unicore64-kernel.tmpl b/Documentation/DocBook/unicore64-kernel.tmpl
index d4b30dc..03899c3 100644
--- a/Documentation/DocBook/unicore64-kernel.tmpl
+++ b/Documentation/DocBook/unicore64-kernel.tmpl
@@ -125,6 +125,13 @@
 		</screen>
 !Parch/unicore64/include/asm/irqflags.h IRQFLAGS_H
 	</sect1>
+	<sect1 id="sect3_2">
+		<title>LD script</title>
+		<screen>
+			See file: arch/unicore64/kernel/vmlinux.lds.S
+		</screen>
+!Parch/unicore64/kernel/vmlinux.lds.S VMLINUX_LDS_S
+	</sect1>
 </chapter>
 
 <chapter id="chapter4">
-- 
1.7.9.5

