From ca35b36d22638f01623deeabe10ed7583288f907 Mon Sep 17 00:00:00 2001
From: Liang Rongrong <liangrongrong@mprc.pku.edu.cn>
Date: Wed, 11 Jan 2012 16:24:52 +0800
Subject: [PATCH 190/641] UniCore64: Add push_lr and pop_lr definitions

Signed-off-by: Liang Rongrong <liangrongrong@mprc.pku.edu.cn>
---
 arch/unicore64/include/arch/asm-common.h |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/unicore64/include/arch/asm-common.h b/arch/unicore64/include/arch/asm-common.h
index 13d81f0..2b60dd8 100644
--- a/arch/unicore64/include/arch/asm-common.h
+++ b/arch/unicore64/include/arch/asm-common.h
@@ -5,6 +5,8 @@
 
 #define csub		cmpsub
 #define dcsub		dcmpsub
+#define push_lr		std.w	lr, [sp-], #8
+#define pop_lr		ldd.w	lr, [sp]+, #8
 
 /*
  * __ASMMACRO_WRAP is ONLY used for assembly macro definitions.
-- 
1.7.9.5

