From 9e0546b091a56ae6a2a4b0d1a1b79823477df647 Mon Sep 17 00:00:00 2001
From: Liang Rongrong <liangrongrong@mprc.pku.edu.cn>
Date: Wed, 11 Jan 2012 16:25:02 +0800
Subject: [PATCH 191/641] UniCore64: Add push_lr and pop_lr in head.S

Signed-off-by: Liang Rongrong <liangrongrong@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/head.S |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/unicore64/kernel/head.S b/arch/unicore64/kernel/head.S
index fa69e960..0eb914e 100644
--- a/arch/unicore64/kernel/head.S
+++ b/arch/unicore64/kernel/head.S
@@ -63,6 +63,7 @@ __priv_sp:
  * it will map the first 64MB of kernel image
  */
 ENTRY(__head_pgtable_init)
+	push_lr
 	/*
 	 * Step one: clear 4k level-one page table for pgd
 	 */
@@ -116,5 +117,6 @@ ENTRY(__head_pgtable_init)
 	dmovl	r0, #UC64_PM_PGTABLE_PGD
 	uc64_set_pgd	r0
 
+	pop_lr
 	return
 ENDPROC(__head_pgtable_init)
-- 
1.7.9.5

