From 2d8bd8cc80101651d4097c19ab6c743ffadd6733 Mon Sep 17 00:00:00 2001
From: Liu Guoli <liuguoli@mprc.pku.edu.cn>
Date: Thu, 12 Jan 2012 18:17:02 +0800
Subject: [PATCH 200/641] UniCore64: Select CONFIG_NO_BOOTMEM

Signed-off-by: Liu Guoli <liuguoli@mprc.pku.edu.cn>
---
 arch/unicore64/Kconfig |    1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/unicore64/Kconfig b/arch/unicore64/Kconfig
index b9ee813..69c42aa 100644
--- a/arch/unicore64/Kconfig
+++ b/arch/unicore64/Kconfig
@@ -10,6 +10,7 @@ config UNICORE64
 	select GENERIC_IRQ_SHOW
 	select OF
 	select OF_EARLY_FLATTREE
+	select NO_BOOTMEM
 	help
 	  UniCore-64 is 64-bit Instruction Set Architecture,
 	  including a series of low-power-consumption RISC chip
-- 
1.7.9.5

