From e65e9e3014884ff95d036f79e651391abd68e723 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Sat, 14 Jan 2012 10:38:40 +0800
Subject: [PATCH 205/641] UniCore64: remove unused header from asm/prom.h

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/include/asm/prom.h |    2 --
 1 file changed, 2 deletions(-)

diff --git a/arch/unicore64/include/asm/prom.h b/arch/unicore64/include/asm/prom.h
index cef81fe..a9e2343 100644
--- a/arch/unicore64/include/asm/prom.h
+++ b/arch/unicore64/include/asm/prom.h
@@ -1,6 +1,4 @@
 #ifndef __UNICORE64_ASM_PROM_H__
 #define __UNICORE64_ASM_PROM_H__
 
-#include <asm/setup.h>
-
 #endif /* __UNICORE64_ASM_PROM_H__ */
-- 
1.7.9.5

