From cd72170d82f364b3d785dd291b4c0b2bc8f22b02 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Mon, 16 Jan 2012 09:20:54 +0800
Subject: [PATCH 217/641] UniCore64: add comments for bitfield.h

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/include/arch/bitfield.h |   16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)

diff --git a/arch/unicore64/include/arch/bitfield.h b/arch/unicore64/include/arch/bitfield.h
index 9fcf604..cd5bc53 100644
--- a/arch/unicore64/include/arch/bitfield.h
+++ b/arch/unicore64/include/arch/bitfield.h
@@ -8,12 +8,16 @@
 
 #include <linux/const.h>
 
-#define __BP(vbits)			(_AC(1, UL) << (vbits))
-#define __BM(vmask, vshift)		(((__BP(vmask)) - 1) << (vshift))
-#define __BS(vmask, vshift)	(~__BM(vmask, vshift))
-
-#define __BF(val, vmask, vshift)		\
-		(((val) << (vshift)) & __BM(vmask, vshift))
+/*
+ * __BP: power of 2 for _len
+ * __BM: mask for bits of _len length from _sft bits
+ * __BS: select mask for bits of _len length from _sft bits
+ * __BF: field value is _val, for bits of _len length from _sft bits
+ */
+#define __BP(_len)		(_AC(1, UL) << (_len))
+#define __BM(_len, _sft)	(((__BP(_len)) - 1) << (_sft))
+#define __BS(_len, _sft)	(~__BM(_len, _sft))
+#define __BF(_val, _len, _sft)	(((_val) << (_sft)) & __BM(_len, _sft))
 
 /*
  * To define 64-bit addresses:
-- 
1.7.9.5

