From 0b39791feb82782dda735380a3da28418f5c88b1 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Tue, 17 Jan 2012 10:40:07 +0800
Subject: [PATCH 228/641] UniCore64: adjust header in uncompress.h

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/boot/uncompress.h |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/unicore64/boot/uncompress.h b/arch/unicore64/boot/uncompress.h
index cb28180..759b5d3 100644
--- a/arch/unicore64/boot/uncompress.h
+++ b/arch/unicore64/boot/uncompress.h
@@ -1,9 +1,10 @@
 #ifndef __UNICORE64_BOOT_UNCOMPRESS_H__
 #define __UNICORE64_BOOT_UNCOMPRESS_H__
 
+#include <arch/asm-debug.h>
+
 extern char input_data[];
 extern char input_data_end[];
-extern void uc64_debug_puts(char *);
 
 #define STATIC static
 #define STATIC_RW_DATA	/* non-static please */
-- 
1.7.9.5

