From c5f02c44cbdea061e6f91ae281c297cab9dfc52f Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Tue, 17 Jan 2012 16:23:18 +0800
Subject: [PATCH 233/641] UniCore64: use boot.lds as load-script  of boot.bin

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/boot/Makefile |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/unicore64/boot/Makefile b/arch/unicore64/boot/Makefile
index c3b55cc..ca42b70 100644
--- a/arch/unicore64/boot/Makefile
+++ b/arch/unicore64/boot/Makefile
@@ -3,7 +3,8 @@ targets		:= Image piggy.bin boot.lds boot.bin zImage unicore64-fpga.dtb
 
 # Do not recognize built-in functions in uncompress.c
 ccflags-y	:= -fno-builtin
-
+# Use boot.lds as load-script
+ldflags-y	:= -T $(obj)/boot.lds
 # Only copy text segment from boot.bin to zImage
 OBJCOPYFLAGS	:= -O binary -S
 
-- 
1.7.9.5

