From 3e2d141fbc220d395abbf27f1935c2587806b7a3 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Tue, 17 Jan 2012 23:23:54 +0800
Subject: [PATCH 238/641] UniCore64: add 8-byte alignment for dtb data

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/boot/boot-head.S |    1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/unicore64/boot/boot-head.S b/arch/unicore64/boot/boot-head.S
index fe33f71..1ac509f 100644
--- a/arch/unicore64/boot/boot-head.S
+++ b/arch/unicore64/boot/boot-head.S
@@ -71,6 +71,7 @@ _start:
 	dmovl	r0, #UC64_VM_KIMAGE_START
 	jump	r0
 
+	__epip4d_align64
 dtb_data:
 	.incbin "arch/unicore64/boot/unicore64-fpga.dtb"
 dtb_data_end:
-- 
1.7.9.5

