From 2d88b174a69d2c1cbcd82db23e18522af338ca00 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Tue, 17 Jan 2012 23:24:25 +0800
Subject: [PATCH 239/641] UniCore64: add cacheline alignment for
 __vectors_table

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/entry.S |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/unicore64/kernel/entry.S b/arch/unicore64/kernel/entry.S
index 56a3aa5..aa0197f 100644
--- a/arch/unicore64/kernel/entry.S
+++ b/arch/unicore64/kernel/entry.S
@@ -1,4 +1,5 @@
 #include <linux/linkage.h>
+#include <asm/cache.h>
 #include <arch/asm-common.h>
 #include <arch/asm-debug.h>
 
@@ -37,6 +38,7 @@ ENTRY(__switch_to)
 	/* FIXME: */
 ENDPROC(__switch_to)
 
+	.align	L1_CACHE_SHIFT
 ENTRY(__vectors_table)
 	call	__vec_invalid	/* 0x00: RESET */
 	call	__vec_invalid	/* 0x04: EEXTN */
-- 
1.7.9.5

