From 294e11b72a7752725cc374fddca6db25e75514c5 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Wed, 18 Jan 2012 10:24:15 +0800
Subject: [PATCH 245/641] UniCore64: put dtb and piggy to the end of the
 zImage

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/boot/boot.S |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/unicore64/boot/boot.S b/arch/unicore64/boot/boot.S
index 4ab1dd3..fa8d87b 100644
--- a/arch/unicore64/boot/boot.S
+++ b/arch/unicore64/boot/boot.S
@@ -71,11 +71,13 @@ _start:
 	dmovl	r0, #UC64_VM_KIMAGE_START
 	jump	r0
 
+	.section .rodata.dtb
 	__epip4d_align64
 dtb_data:
 	.incbin "arch/unicore64/boot/unicore64-fpga.dtb"
 dtb_data_end:
 
+	.section .rodata.piggy
 	__epip4d_align64
 	.globl	input_data
 input_data:
-- 
1.7.9.5

