From 3341290cc2bb07e22b2ea11bb43cc25dac084d5c Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Mon, 30 Jan 2012 17:26:23 +0800
Subject: [PATCH 287/641] UniCore64: adjust struct format of __trap_info

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/mm/fault.c |   64 ++++++++++++++++++++++-----------------------
 1 file changed, 32 insertions(+), 32 deletions(-)

diff --git a/arch/unicore64/mm/fault.c b/arch/unicore64/mm/fault.c
index 63734a6..2e71e44 100644
--- a/arch/unicore64/mm/fault.c
+++ b/arch/unicore64/mm/fault.c
@@ -42,41 +42,41 @@ struct __trap_info {
 };
 
 static struct __trap_info __itrap_info[] = {
-	{ NULL,	"0: no error"				},
-	{ NULL,	"1: access debug-specific area"		},
-	{ NULL,	"2: 2nd level pagetable non-exist"	},
-	{ NULL,	"3: 3rd level pagetable non-exist"	},
-	{ __do_pagefault,	"4: page non-exist"	},
-	{ NULL,	"5: superpage non-exist"		},
-	{ NULL,	"6: non-executable instruction"		},
-	{ NULL,	"7: access US=0 page in user mode"	},
-	{ NULL,	"8: memory access error"		},
-	{ NULL,	"9: icache parity error"		},
-	{ NULL,	"a: icache access error"		},
-	{ NULL,	"b: illegal instruction"		},
-	{ NULL,	"c: previlege instruction"		},
-	{ NULL,	"d: eret instruction exception"		},
-	{ NULL,	"e: unknown itrap fault"		},
-	{ NULL,	"f: unknown itrap fault"		},
+	{ NULL,			"0: no error"				},
+	{ NULL,			"1: access debug-specific area"		},
+	{ NULL,			"2: 2nd level pagetable non-exist"	},
+	{ NULL,			"3: 3rd level pagetable non-exist"	},
+	{ __do_pagefault,	"4: page non-exist"			},
+	{ NULL,			"5: superpage non-exist"		},
+	{ NULL,			"6: non-executable instruction"		},
+	{ NULL,			"7: access US=0 page in user mode"	},
+	{ NULL,			"8: memory access error"		},
+	{ NULL,			"9: icache parity error"		},
+	{ NULL,			"a: icache access error"		},
+	{ NULL,			"b: illegal instruction"		},
+	{ NULL,			"c: previlege instruction"		},
+	{ NULL,			"d: eret instruction exception"		},
+	{ NULL,			"e: unknown itrap fault"		},
+	{ NULL,			"f: unknown itrap fault"		},
 };
 
 static struct __trap_info __dtrap_info[] = {
-	{ NULL,	"0: no error"				},
-	{ NULL,	"1: access debug-specific area"		},
-	{ NULL,	"2: 2nd level pagetable non-exist"	},
-	{ NULL,	"3: 3rd level pagetable non-exist"	},
-	{ __do_pagefault,	"4: page non-exist"	},
-	{ NULL,	"5: superpage non-exist"		},
-	{ NULL,	"6: read unreadable address"		},
-	{ NULL,	"7: write unwritable address"		},
-	{ NULL,	"8: access US=0 page in user mode"	},
-	{ NULL,	"9: unalignment access"			},
-	{ NULL,	"a: memory access error"		},
-	{ NULL,	"b: dcache parity error"		},
-	{ NULL,	"c: dcache access error"		},
-	{ NULL,	"d: divide zero"			},
-	{ NULL,	"e: access cacheable area in atomic op"	},
-	{ NULL,	"f: unknown dtrap fault"		},
+	{ NULL,			"0: no error"				},
+	{ NULL,			"1: access debug-specific area"		},
+	{ NULL,			"2: 2nd level pagetable non-exist"	},
+	{ NULL,			"3: 3rd level pagetable non-exist"	},
+	{ __do_pagefault,	"4: page non-exist"			},
+	{ NULL,			"5: superpage non-exist"		},
+	{ NULL,			"6: read unreadable address"		},
+	{ NULL,			"7: write unwritable address"		},
+	{ NULL,			"8: access US=0 page in user mode"	},
+	{ NULL,			"9: unalignment access"			},
+	{ NULL,			"a: memory access error"		},
+	{ NULL,			"b: dcache parity error"		},
+	{ NULL,			"c: dcache access error"		},
+	{ NULL,			"d: divide zero"			},
+	{ NULL,			"e: access cacheable area in atomic op"	},
+	{ NULL,			"f: unknown dtrap fault"		},
 };
 
 void __do_itrap(unsigned long addr, struct pt_regs *regs)
-- 
1.7.9.5

