From 26f6c14343c8015c4a4de3e5aab266c7910a3fc1 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Tue, 31 Jan 2012 10:38:39 +0800
Subject: [PATCH 292/641] UniCore64: modify comment in boot.lds.S

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/boot/boot.lds.S |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/unicore64/boot/boot.lds.S b/arch/unicore64/boot/boot.lds.S
index 8679e2a..31e2ada 100644
--- a/arch/unicore64/boot/boot.lds.S
+++ b/arch/unicore64/boot/boot.lds.S
@@ -15,7 +15,7 @@ SECTIONS
 
 	. = ALIGN(8);
 
-	/* epip4d: trick to clear bss */
+	/* ____bss_temp_handler: a trick to clear bss */
 	. = UC64_PM_ZIMAGE_HEAP_END;
 	__bss_start = UC64_PM_ZIMAGE_HEAP_END;
 	.bss : { *(.bss) }
-- 
1.7.9.5

