From 174bde8c2aae77cba507cbd141ccb6cbe129a546 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Wed, 1 Feb 2012 10:15:09 +0800
Subject: [PATCH 293/641] UniCore64: add comment for debug_putx

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/lib/debug.S |    1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/unicore64/lib/debug.S b/arch/unicore64/lib/debug.S
index 86527e4..9021f5f 100644
--- a/arch/unicore64/lib/debug.S
+++ b/arch/unicore64/lib/debug.S
@@ -38,6 +38,7 @@ ENDPROC(uc64_debug_puts)
 ENTRY(uc64_debug_putx)
 	__putdata	r0, r14, r15
 
+	/* ____debug_putx: we add a blank after data */
 	mov	r0, #0x20
 	__putchar	r0, r14, r15
 
-- 
1.7.9.5

