From 89386e2d2424b85736cc170f53625908b978ee00 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Wed, 1 Feb 2012 10:18:48 +0800
Subject: [PATCH 294/641] UniCore64: remove obsolete __putchar function

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/include/arch/asm-debug.h |    5 -----
 1 file changed, 5 deletions(-)

diff --git a/arch/unicore64/include/arch/asm-debug.h b/arch/unicore64/include/arch/asm-debug.h
index de3b29e..d0b81ac 100644
--- a/arch/unicore64/include/arch/asm-debug.h
+++ b/arch/unicore64/include/arch/asm-debug.h
@@ -28,11 +28,6 @@ __ASMMACRO_WRAP(.macro	__putdata, rdata, rt1, rt2;
 			mov	r0, &rt1;
 			mov	r1, &rt2;
 		.endm)
-#else
-/* For RTL simulation environment */
-__ASMMACRO_WRAP(.macro	__putchar, rchar, rt1, rt2;
-			movc	p6.c0, &rchar, #1;
-		.endm)
 #endif /* CONFIG_DEBUG_OCD */
 
 #ifndef __ASSEMBLY__
-- 
1.7.9.5

