From 867e1dc79b649ddc6447993fa6d577f4d1a1d677 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Fri, 3 Feb 2012 09:26:58 +0800
Subject: [PATCH 297/641] UniCore64: correct sp value before mmu enabled

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/head.S |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/arch/unicore64/kernel/head.S b/arch/unicore64/kernel/head.S
index c5cdff2..d21fee4 100644
--- a/arch/unicore64/kernel/head.S
+++ b/arch/unicore64/kernel/head.S
@@ -32,12 +32,11 @@ ENTRY(stext)
 	__head_initialize	r0
 	__cache_initialize	r0
 
-	ldd	sp, __priv_sp
-
 	/*
 	 * Enable the MMU.  This completely changes the structure of the visible
 	 * memory space.  You will not be able to trace execution through this.
 	 */
+	dmovl	sp, #UC64_PM_ZIMAGE_STACKTOP
 	call	__head_pgtable_init
 
 	__invalid_tlb		r0
@@ -46,6 +45,8 @@ ENTRY(stext)
 	dor	r0, r0, #CP0_CTRLREG_MMU
 	movc	CP0_CTRLREG, r0, #0
 
+	ldd	sp, __priv_sp
+
 	__head_clear_bss
 
 	b	start_kernel
-- 
1.7.9.5

