From 4311261557b896efe48ba312f157de2dad2926e1 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Thu, 16 Feb 2012 14:27:38 +0800
Subject: [PATCH 302/641] UniCore64: align all segments to cache line

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/vmlinux.lds.S |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/unicore64/kernel/vmlinux.lds.S b/arch/unicore64/kernel/vmlinux.lds.S
index 39f39e8..4f1c885 100644
--- a/arch/unicore64/kernel/vmlinux.lds.S
+++ b/arch/unicore64/kernel/vmlinux.lds.S
@@ -22,7 +22,7 @@ SECTIONS
 	__init_begin = .;
 	HEAD_TEXT_SECTION
 	INIT_TEXT_SECTION(PAGE_SIZE)
-	INIT_DATA_SECTION(16)
+	INIT_DATA_SECTION(L1_CACHE_BYTES)
 	PERCPU_SECTION(L1_CACHE_BYTES)
 	__init_end = .;
 
@@ -45,7 +45,7 @@ SECTIONS
 	EXCEPTION_TABLE(L1_CACHE_BYTES)
 	NOTES
 
-	BSS_SECTION(0, 0, 0)
+	BSS_SECTION(L1_CACHE_BYTES, L1_CACHE_BYTES, L1_CACHE_BYTES)
 	_end = .;
 
 	STABS_DEBUG
-- 
1.7.9.5

