From ed59d0cdc5b266a3f3d5a274fb2eebec32523144 Mon Sep 17 00:00:00 2001
From: Liang Rongrong <liangrongrong@mprc.pku.edu.cn>
Date: Mon, 20 Feb 2012 14:42:02 +0800
Subject: [PATCH 303/641] UniCore64: Add superpage direct-mapping support

Signed-off-by: Liang Rongrong <liangrongrong@mprc.pku.edu.cn>
---
 arch/unicore64/mm/mmu.c |   18 ++++++++++--------
 1 file changed, 10 insertions(+), 8 deletions(-)

diff --git a/arch/unicore64/mm/mmu.c b/arch/unicore64/mm/mmu.c
index c7e0297..8804047 100644
--- a/arch/unicore64/mm/mmu.c
+++ b/arch/unicore64/mm/mmu.c
@@ -14,20 +14,22 @@ static void __init uc64_create_direct_mapping(phys_addr_t start,
 		unsigned long length)
 {
 	phys_addr_t phys;
-	pgprot_t prot_pud;
-	pud_t *pud;
+	phys_addr_t phys_pud = UC64_PM_PGTABLE_PUD_DM00;
+	pgprot_t prot_pmd;
+	pmd_t *pmd;
 
 	/* ONLY SUPERPAGE supported */
-	if (start | length | UC64_PMD_MASK)
+	if ((start | length) & ~UC64_PMD_MASK)
 		BUG();
 
-	prot_pud = __pgprot(UC64_PMD_TYPE_CACHE | UC64_PMD_EXIST
-			| UC64_PMD_SPAGE);
-	pud = pud_offset(pgd_offset_k(start), (unsigned long)__va(start));
+	/* the first pud has been written in head.S */
+	prot_pmd = __pgprot(UC64_PMD_TYPE_CACHE | UC64_PMD_EXIST
+			| UC64_PMD_RWX | UC64_PMD_SPAGE);
+	pmd = phys_pud + pmd_index((unsigned long)__va(start));
 
 	for (phys = start; phys < (start + length); phys += UC64_PMD_SIZE) {
-		set_pud(pud, __pud(phys | pgprot_val(prot_pud)));
-		pud++;
+		set_pmd(pmd, __pmd(phys | pgprot_val(prot_pmd)));
+		pmd++;
 	}
 }
 
-- 
1.7.9.5

