From c14838b4f84d5b30cea1bda00ef4b8580e443929 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Fri, 2 Mar 2012 17:14:06 +0800
Subject: [PATCH 315/641] UniCore64: add zone_sizes_init and max_pfn_init
 function

Signed-off-by: Liu Guoli <liuguoli@mprc.pku.edu.cn>
Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/mm/init.c |   34 +++++++++++++++++++++++++++++++++-
 1 file changed, 33 insertions(+), 1 deletion(-)

diff --git a/arch/unicore64/mm/init.c b/arch/unicore64/mm/init.c
index af98387..1ca1125 100644
--- a/arch/unicore64/mm/init.c
+++ b/arch/unicore64/mm/init.c
@@ -1,5 +1,6 @@
 #include <linux/kernel.h>
 #include <linux/memblock.h>
+#include <linux/bootmem.h>
 
 #include <asm/sections.h>
 #include <asm/setup_arch.h>
@@ -22,7 +23,19 @@ void free_initmem(void)
 	BUG();
 }
 
-void __init setup_arch_memory(void)
+static void __init zone_sizes_init(void)
+{
+	unsigned long zone_sizes[MAX_NR_ZONES];
+
+	/* Initialize each of zones. */
+	memset(zone_sizes, 0, sizeof(zone_sizes));
+	zone_sizes[ZONE_NORMAL] = max_pfn;
+
+	/* Initialize mem_map[] */
+	free_area_init(zone_sizes);
+}
+
+static void __init memblock_init(void)
 {
 	/* Reserve the kernel text, kernel data and initrd with memblock. */
 	memblock_reserve(__pa(_text), _end - _text);
@@ -32,6 +45,25 @@ void __init setup_arch_memory(void)
 
 	memblock_allow_resize();
 	memblock_dump_all();
+}
 
+static void __init max_pfn_init(void)
+{
+	struct memblock_region *reg;
+	unsigned long last_pfn;
+
+	/* Find the PFN for the last page in the system. */
+	for_each_memblock(memory, reg) {
+		last_pfn = memblock_region_memory_end_pfn(reg);
+		if (max_pfn < last_pfn)
+			max_pfn = last_pfn;
+	}
+}
+
+void __init setup_arch_memory(void)
+{
+	memblock_init();
+	max_pfn_init();
 	paging_init();
+	zone_sizes_init();
 }
-- 
1.7.9.5

