From 4a315f0b63a4e5d501128cf9cd99e8359b9ed436 Mon Sep 17 00:00:00 2001
From: Liang Rongrong <liangrongrong@mprc.pku.edu.cn>
Date: Wed, 7 Mar 2012 11:05:02 +0800
Subject: [PATCH 321/641] UniCore64: Add free_all_bootmem() and max_mapnr

Signed-off-by: Liang Rongrong <liangrongrong@mprc.pku.edu.cn>
---
 arch/unicore64/mm/init.c |    8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/arch/unicore64/mm/init.c b/arch/unicore64/mm/init.c
index 1ca1125..e0a2ed4 100644
--- a/arch/unicore64/mm/init.c
+++ b/arch/unicore64/mm/init.c
@@ -6,12 +6,14 @@
 #include <asm/setup_arch.h>
 
 /**
- * mem_init() -
+ * mem_init() - release free pages to the buddy allocator
+ * Returns the number of pages actually released.
  */
 void __init mem_init(void)
 {
-	/* FIXME */
-	BUG();
+	max_mapnr = pfn_to_page(max_pfn) - mem_map;
+
+	totalram_pages += free_all_bootmem();
 }
 
 /**
-- 
1.7.9.5

