From 636a2810d7ae860eca17c94bf527699dc2e1b041 Mon Sep 17 00:00:00 2001
From: caosong <caosong@mprc.pku.edu.cn>
Date: Thu, 29 Mar 2012 10:45:01 +0800
Subject: [PATCH 328/641] UniCore64: set arguments for itrap main handler

Signed-off-by: caosong <caosong@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/entry.S |   20 +++++++++++++++++++-
 1 file changed, 19 insertions(+), 1 deletion(-)

diff --git a/arch/unicore64/kernel/entry.S b/arch/unicore64/kernel/entry.S
index f1a92cf..bc1542d 100644
--- a/arch/unicore64/kernel/entry.S
+++ b/arch/unicore64/kernel/entry.S
@@ -65,10 +65,28 @@ ENDPROC(__vec_invalid)
 ENTRY(__vec_itrap)
 	__priv_context_save
 
+	@
+	@ set args, then call itrap main handler
+	@
+	@ r0 - address of faulting instruction
+	@ r1 - pointer to registers on stack
+	@
+	movc	r0, CP0_EPC, #1
+	dmov	r1, sp
+
+	@
+	@ enable interrupts by a temporary register r5
+	@
+	enable_irq r5
+
 	call	__do_itrap
 
+	@
+	@ disable interrupts before pulling preserved data off the stack
+	@
+	disable_irq r5
+
 	__priv_context_restore
-	eret
 ENDPROC(__vec_itrap)
 
 ENTRY(__vec_dtrap)
-- 
1.7.9.5

