From 321452a310c70654c80eace583b68e4c9da0e394 Mon Sep 17 00:00:00 2001
From: caosong <caosong@mprc.pku.edu.cn>
Date: Thu, 29 Mar 2012 11:28:15 +0800
Subject: [PATCH 329/641] UniCore64: set arguments for dtrap main handler

Signed-off-by: caosong <caosong@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/entry.S |   20 +++++++++++++++++++-
 1 file changed, 19 insertions(+), 1 deletion(-)

diff --git a/arch/unicore64/kernel/entry.S b/arch/unicore64/kernel/entry.S
index bc1542d..3cd5a15 100644
--- a/arch/unicore64/kernel/entry.S
+++ b/arch/unicore64/kernel/entry.S
@@ -92,10 +92,28 @@ ENDPROC(__vec_itrap)
 ENTRY(__vec_dtrap)
 	__priv_context_save
 
+	@
+	@ set args, then call itrap main handler
+	@
+	@ r0 - address of faulting instruction
+	@ r1 - pointer to registers on stack
+	@
+	movc	r0, CP0_EPC, #1
+	dmov	r1, sp
+
+	@
+	@ enable interrupts by a temporary register r5
+	@
+	enable_irq r5
+
 	call	__do_dtrap
 
+	@
+	@ disable interrupts before pulling preserved data off the stack
+	@
+	disable_irq r5
+
 	__priv_context_restore
-	eret
 ENDPROC(__vec_dtrap)
 
 ENTRY(__vec_int_puv3)
-- 
1.7.9.5

