From 2a0758cca4a545162d792520ba49038c4d2996af Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Fri, 30 Mar 2012 15:14:55 +0800
Subject: [PATCH 330/641] UniCore64: correct dtrap entry code

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/entry.S |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/unicore64/kernel/entry.S b/arch/unicore64/kernel/entry.S
index 3cd5a15..93478a9 100644
--- a/arch/unicore64/kernel/entry.S
+++ b/arch/unicore64/kernel/entry.S
@@ -93,12 +93,12 @@ ENTRY(__vec_dtrap)
 	__priv_context_save
 
 	@
-	@ set args, then call itrap main handler
+	@ set args, then call dtrap main handler
 	@
-	@ r0 - address of faulting instruction
+	@ r0 - address of faulting address
 	@ r1 - pointer to registers on stack
 	@
-	movc	r0, CP0_EPC, #1
+	movc	r0, CP0_EPC, #0
 	dmov	r1, sp
 
 	@
-- 
1.7.9.5

