From c80b8303645e63599ea2e0f2826359b0c3aace98 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Sat, 31 Mar 2012 16:33:59 +0800
Subject: [PATCH 334/641] UniCore64: remove setup_arch_resource()

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/setup.c |    7 -------
 1 file changed, 7 deletions(-)

diff --git a/arch/unicore64/kernel/setup.c b/arch/unicore64/kernel/setup.c
index 7183aa5..3fe81fd 100644
--- a/arch/unicore64/kernel/setup.c
+++ b/arch/unicore64/kernel/setup.c
@@ -36,12 +36,6 @@ static void __init setup_arch_param(char **cmdline_p)
 	*cmdline_p = builtin_cmdline;
 }
 
-static void __init setup_arch_resource(void)
-{
-	/* FIXME */
-	BUG();
-}
-
 /**
  * setup_arch() - Architecture-specific boot-time initializations
  * @cmdline_p:
@@ -56,5 +50,4 @@ void __init setup_arch(char **cmdline_p)
 	setup_arch_param(cmdline_p);
 	setup_arch_memory();
 	unflatten_device_tree();
-	setup_arch_resource();
 }
-- 
1.7.9.5

