From fa4b7fdb2df9af02ae7266df376697f441d99c1e Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Wed, 16 May 2012 16:18:46 +0800
Subject: [PATCH 343/641] UniCore64: adjust delta2ns calculations in itimer

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/time.c |    7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/arch/unicore64/kernel/time.c b/arch/unicore64/kernel/time.c
index eb99706..6cd6ac8 100644
--- a/arch/unicore64/kernel/time.c
+++ b/arch/unicore64/kernel/time.c
@@ -17,7 +17,8 @@
 #define __itimer_irq_clear()					\
 	__write_cp(__read_cp(CP0_INTR) & ~CP0_INTR_ITM, CP0_INTR)
 
-#define MIN_COUNTER_DELTA			(2)
+#define MIN_COUNTER_DELTA		(USEC_PER_SEC / CONFIG_HZ)
+#define MAX_COUNTER_DELTA		(NSEC_PER_SEC / CONFIG_HZ)
 
 static cycle_t __itimer_read(struct clocksource *cs)
 {
@@ -89,9 +90,9 @@ void __init time_init(void)
 	clockevents_calc_mult_shift(&__itimer_ce, CLOCK_TICK_RATE, 5);
 
 	__itimer_ce.max_delta_ns =
-		clockevent_delta2ns(CLOCK_TICK_RATE, &__itimer_ce);
+		clockevent_delta2ns(MAX_COUNTER_DELTA, &__itimer_ce);
 	__itimer_ce.min_delta_ns =
-		clockevent_delta2ns(MIN_COUNTER_DELTA * 2, &__itimer_ce) + 1;
+		clockevent_delta2ns(MIN_COUNTER_DELTA, &__itimer_ce);
 
 	clocksource_register_hz(&__itimer_cs, CLOCK_TICK_RATE);
 	clockevents_register_device(&__itimer_ce);
-- 
1.7.9.5

