From 97d01928947b4727c3cd4de63cd9d4f1152053d3 Mon Sep 17 00:00:00 2001
From: caosong <caosong@mprc.pku.edu.cn>
Date: Thu, 17 May 2012 16:01:04 +0800
Subject: [PATCH 349/641] UniCore64: Add clean DOC_DIR into archclean

Signed-off-by: caosong <caosong@mprc.pku.edu.cn>
---
 arch/unicore64/Makefile |    1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/unicore64/Makefile b/arch/unicore64/Makefile
index a5d7ef9..0842ce1 100644
--- a/arch/unicore64/Makefile
+++ b/arch/unicore64/Makefile
@@ -37,6 +37,7 @@ zImage: vmlinux
 
 archclean:
 	$(Q)$(MAKE) $(clean)=$(BOOT_DIR)
+	$(Q)$(MAKE) $(clean)=$(DOC_DIR)
 
 archdoc:
 	$(Q)$(MAKE) $(build)=scripts build_docproc
-- 
1.7.9.5

