From fc4632ce434c87346388544aa23c69025fe05064 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Thu, 31 May 2012 14:30:03 +0800
Subject: [PATCH 354/641] UniCore64: adjust code format for uc64_debug_putx

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/lib/debug.S |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/unicore64/lib/debug.S b/arch/unicore64/lib/debug.S
index 0a089d6..8136e92 100644
--- a/arch/unicore64/lib/debug.S
+++ b/arch/unicore64/lib/debug.S
@@ -44,7 +44,7 @@ ENTRY(uc64_debug_putx)
 	__putdata	r0
 
 	/* ____debug_putx: we add a blank after data */
-	mov	r0, #0x20
+	mov		r0, #0x20
 	__putchar	r0
 
 	return
-- 
1.7.9.5

