From 5164888e4e24bf46cde95565ae89ce523f6993bb Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Tue, 5 Jun 2012 10:03:08 +0800
Subject: [PATCH 365/641] UniCore64: correct the comment in ocd_console.c

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/lib/ocd_console.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/unicore64/lib/ocd_console.c b/arch/unicore64/lib/ocd_console.c
index ce9d8a8..85c83e9 100644
--- a/arch/unicore64/lib/ocd_console.c
+++ b/arch/unicore64/lib/ocd_console.c
@@ -38,7 +38,8 @@ static void ocd_con_putcs(struct vc_data *vc, const unsigned short *s,
 	char c = '\n';
 
 	/* For each element in *s (16bit in an unsigned short type):
-	 * High 8-bit is background color, low 8-bit is foreground char */
+	 * High 8-bit is color, low 8-bit is char,
+	 * and only char is useful here. */
 	while (*s && (n > 0)) {
 		uc64_debug_puts((char *)s, 1);
 		s++;
-- 
1.7.9.5

