From 51006ceb8b305cd973d0df6acb16b07471fd9393 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Fri, 8 Jun 2012 09:09:22 +0800
Subject: [PATCH 371/641] UniCore64: unalignment sticky is unlikely happened

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/mm/fault.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/unicore64/mm/fault.c b/arch/unicore64/mm/fault.c
index 020c966..6fdeead 100644
--- a/arch/unicore64/mm/fault.c
+++ b/arch/unicore64/mm/fault.c
@@ -99,7 +99,7 @@ void __do_dtrap(unsigned long addr, struct pt_regs *regs)
 	int dtno;
 
 	dtno = __dtrap_stat();
-	if (dtno & CP0_TRAPSTAT_UNALIGN) {
+	if (unlikely(dtno & CP0_TRAPSTAT_UNALIGN)) {
 		pr_err("Error: DTRAP occured with Un-Alignment Sticky!\n");
 		dtno &= CP0_TRAPSTAT_SELECT;
 	}
-- 
1.7.9.5

