From 26403410eb689a352b72eff457ba47517da6c830 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Tue, 12 Jun 2012 08:47:27 +0800
Subject: [PATCH 374/641] UniCore64: remove unused macros in mmuops

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/include/arch/asm-mmuops.h |   11 -----------
 1 file changed, 11 deletions(-)

diff --git a/arch/unicore64/include/arch/asm-mmuops.h b/arch/unicore64/include/arch/asm-mmuops.h
index abe8abc..e5e5e6a 100644
--- a/arch/unicore64/include/arch/asm-mmuops.h
+++ b/arch/unicore64/include/arch/asm-mmuops.h
@@ -52,15 +52,4 @@ __ASMMACRO_WRAP(.macro	__set_pgd, rpgd;
 			movc	p0.c2, &rpgd, #0;
 		.endm)
 
-__ASMMACRO_WRAP(.macro	__itrap_status, rtrap;
-			movc	&rtrap, p0.c3, #0;
-		.endm)
-
-__ASMMACRO_WRAP(.macro	__dtrap_status, rtrap;
-			movc	&rtrap, p0.c3, #1;
-		.endm)
-
-__ASMMACRO_WRAP(.macro	__dtrap_addr, raddr;
-			movc	&raddr, p0.c4, #0;
-		.endm)
 #endif /* __UNICORE64_ARCH_ASM_MMUOPS_H__ */
-- 
1.7.9.5

