From c6b81ea5b2b0020464171e631646200377243eca Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Thu, 14 Jun 2012 10:14:40 +0800
Subject: [PATCH 385/641] UniCore64: adjust __show_cp0_regs format

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/cpu-uc64.c |   18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/arch/unicore64/kernel/cpu-uc64.c b/arch/unicore64/kernel/cpu-uc64.c
index b6ac561..6b27f64 100644
--- a/arch/unicore64/kernel/cpu-uc64.c
+++ b/arch/unicore64/kernel/cpu-uc64.c
@@ -206,12 +206,12 @@ void __show_uc64_regs(void)
 void __show_cp0_regs(void)
 {
 	pr_emerg("\nCP0 (System Control) Information:\n");
-	pr_emerg(" CPUID:      %16lx", __read_cp_op(CP0_CPUID, 0));
+	pr_emerg(" CPUID:      %16lx\n", __read_cp_op(CP0_CPUID, 0));
 	pr_emerg(" CACHE:      %16lx\n", __read_cp_op(CP0_CPUID, 1));
-	pr_emerg(" CTRLREG:    %16lx", __read_cp(CP0_CTRLREG));
+	pr_emerg(" CTRLREG:    %16lx\n", __read_cp(CP0_CTRLREG));
 	pr_emerg(" TTB/ASID:   %16lx\n", __read_cp(CP0_TTB_ASID));
-	pr_emerg(" ITRAPSTAT:  %16lx", __read_cp_op(CP0_TRAPSTAT, 0));
-	pr_emerg(" DTRAPSTAT:  %16lx", __read_cp_op(CP0_TRAPSTAT, 1));
+	pr_emerg(" ITRAPSTAT:  %16lx\n", __read_cp_op(CP0_TRAPSTAT, 0));
+	pr_emerg(" DTRAPSTAT:  %16lx\n", __read_cp_op(CP0_TRAPSTAT, 1));
 	pr_emerg(" ITRAPADDR:  %16lx\n", __read_cp_op(CP0_TRAPADDR, 1));
 	pr_emerg(" DTRAPADDR:  %16lx\n", __read_cp_op(CP0_TRAPADDR, 0));
 
@@ -236,11 +236,11 @@ void __show_cp0_regs(void)
 			__read_cp_op(CP0_MRAR, 14),
 			__read_cp_op(CP0_MRAR, 15));
 
-	pr_emerg(" VECBASE:    %16lx", __read_cp(CP0_VECBASE));
-	pr_emerg(" INTR_TYPE:  %16lx", __read_cp_op(CP0_INTR, 0));
+	pr_emerg(" VECBASE:    %16lx\n", __read_cp(CP0_VECBASE));
+	pr_emerg(" INTR_TYPE:  %16lx\n", __read_cp_op(CP0_INTR, 0));
 	pr_emerg(" INTR_PRIO:  %16lx\n", __read_cp_op(CP0_INTR, 7));
-	pr_emerg(" INNER_TIMER_COUNTER: %16lx", __read_cp_op(CP0_INTR, 9));
-	pr_emerg(" INNER_TIMER_MATCH:   %16lx", __read_cp_op(CP0_INTR, 10));
+	pr_emerg(" ITM_CNTR:   %16lx\n", __read_cp_op(CP0_INTR, 9));
+	pr_emerg(" ITM_MATCH:  %16lx\n", __read_cp_op(CP0_INTR, 10));
 
-	pr_emerg(" R/W Margin: %16lx", __read_cp(CP0_RWMARGIN));
+	pr_emerg(" R/W Margin: %16lx\n", __read_cp(CP0_RWMARGIN));
 }
-- 
1.7.9.5

