From 2aa2280f8c3da5fdb745280cf8e1c45fe790a2cd Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Wed, 22 Aug 2012 17:08:18 +0800
Subject: [PATCH 398/641] unicore64: Enable __ARCH_WANT_SYSCALL_NO_AT

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/include/asm/unistd.h |    1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/unicore64/include/asm/unistd.h b/arch/unicore64/include/asm/unistd.h
index 0cdaef8..ad33b41 100644
--- a/arch/unicore64/include/asm/unistd.h
+++ b/arch/unicore64/include/asm/unistd.h
@@ -2,6 +2,7 @@
 #define __UNICORE64_ASM_UNISTD_H__
 
 #define __ARCH_WANT_SYNC_FILE_RANGE2
+#define __ARCH_WANT_SYSCALL_NO_AT
 
 /* Use the standard ABI for syscalls. */
 #include <asm-generic/unistd.h>
-- 
1.7.9.5

