From 34fce4d8fbcdb5734f3671aa31a00fd86a63e46c Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Wed, 29 Aug 2012 11:24:55 +0800
Subject: [PATCH 412/641] unicore64: Adjust arch/unicore64/lib/Makefile

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/lib/Makefile |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/unicore64/lib/Makefile b/arch/unicore64/lib/Makefile
index b08f4c3..e98c459 100644
--- a/arch/unicore64/lib/Makefile
+++ b/arch/unicore64/lib/Makefile
@@ -1,5 +1,5 @@
-lib-y			:= delay.o debug.o
-lib-y			+= string-simple.o
-lib-y			+= mcount.o
+lib-y				:= delay.o debug.o
+lib-y				+= string-simple.o
 
+lib-$(CONFIG_DEBUG_MCOUNT)	+= mcount.o
 lib-$(CONFIG_OCD_CONSOLE)	+= ocd_console.o
-- 
1.7.9.5

