From cd8e0062f7067e63c3fac6aabb59e3b60fc4ac4e Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Wed, 5 Sep 2012 17:30:18 +0800
Subject: [PATCH 414/641] unicore64: Add missing TIF_NOTIFY_RESUME

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/include/asm/thread_info.h |    1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/unicore64/include/asm/thread_info.h b/arch/unicore64/include/asm/thread_info.h
index cad9798..ef01c59 100644
--- a/arch/unicore64/include/asm/thread_info.h
+++ b/arch/unicore64/include/asm/thread_info.h
@@ -62,6 +62,7 @@ static inline struct thread_info *current_thread_info(void)
 #define TIF_SYSCALL_TRACE	2 /* syscall trace active */
 #define TIF_RESTORE_SIGMASK	3 /* restore signal mask in do_signal */
 #define TIF_MEMDIE		4 /* is terminating due to OOM killer */
+#define TIF_NOTIFY_RESUME	5 /* callback before returning to user */
 
 #endif /* __KERNEL__ */
 #endif /* __UNICORE64_ASM_THREAD_INFO_H__ */
-- 
1.7.9.5

