From 939e3c32880c62dbb7f060469a6d9dde211ed2bd Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Thu, 6 Sep 2012 10:52:51 +0800
Subject: [PATCH 415/641] unicore64: Make EARLY_PRINTK depend on PRINTK

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/Kconfig.debug |    1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/unicore64/Kconfig.debug b/arch/unicore64/Kconfig.debug
index 87c4a44..02f865e 100644
--- a/arch/unicore64/Kconfig.debug
+++ b/arch/unicore64/Kconfig.debug
@@ -18,6 +18,7 @@ config DEBUG_MCOUNT
 
 config EARLY_PRINTK
 	def_bool DEBUG_OCD
+	depends on PRINTK
 	help
 	  Write kernel log output directly into the ocd or to a serial port.
 
-- 
1.7.9.5

