From efe5bb49f4ee7fecb04ff21a4d03e2f47496f458 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Thu, 6 Sep 2012 19:24:18 +0800
Subject: [PATCH 417/641] unicore64: Add free_initmem

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/mm/init.c |   24 ++++++++++++++++++++++--
 1 file changed, 22 insertions(+), 2 deletions(-)

diff --git a/arch/unicore64/mm/init.c b/arch/unicore64/mm/init.c
index 8b49fd7..d498917 100644
--- a/arch/unicore64/mm/init.c
+++ b/arch/unicore64/mm/init.c
@@ -16,13 +16,33 @@ void __init mem_init(void)
 	totalram_pages += free_all_bootmem();
 }
 
+static inline unsigned long __free_area(unsigned long begin, unsigned long end)
+{
+	struct page *page;
+	unsigned long addr;
+
+	for (addr = begin; addr < end; addr += PAGE_SIZE) {
+		page = virt_to_page(addr);
+		ClearPageReserved(page);
+		init_page_count(page);
+		__free_page(page);
+	}
+
+	return (end - begin) >> PAGE_SHIFT;
+}
+
 /**
  * free_initmem() -
  */
 void free_initmem(void)
 {
-	/* FIXME */
-	BUG();
+	unsigned long pages;
+
+	pages = __free_area((unsigned long)__init_begin,
+			    (unsigned long)__init_end);
+	pr_info("Freeing init memory: %ld pages\n", pages);
+
+	totalram_pages += pages;
 }
 
 #ifdef CONFIG_BLK_DEV_INITRD
-- 
1.7.9.5

