From 3912738f59599ea89ead08ce4e011f09b54a20fb Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Thu, 6 Sep 2012 20:04:44 +0800
Subject: [PATCH 418/641] unicore64: Use print_hex_dump to rewrite __dump_mem

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/showinfo.c |   18 ++----------------
 1 file changed, 2 insertions(+), 16 deletions(-)

diff --git a/arch/unicore64/kernel/showinfo.c b/arch/unicore64/kernel/showinfo.c
index 1b85f86..a5eb5a3 100644
--- a/arch/unicore64/kernel/showinfo.c
+++ b/arch/unicore64/kernel/showinfo.c
@@ -197,23 +197,9 @@ void __show_cp0_regs(void)
 
 static void __dump_mem(unsigned long bottom, unsigned long top)
 {
-	unsigned long p;
-	int i;
-	unsigned int val;
-
 	pr_info("Dump memory from 0x%016lx to 0x%016lx:\n", bottom, top);
-
-	for (p = bottom & ~31; p < top; ) {
-		pr_info(" %04lx:", p & 0xffff);
-
-		for (i = 0; i < 8; i++, p += 4) {
-			if (__get_user(val, (unsigned int *)p)) {
-				pr_info("Fail to access 0x%016lx\n", p);
-				return;
-			}
-			pr_cont(" %08x", val);
-		}
-	}
+	print_hex_dump(KERN_INFO, " ", DUMP_PREFIX_ADDRESS, 16, 4,
+			(void *)bottom, top - bottom, true);
 }
 
 /**
-- 
1.7.9.5

