From 7cee4f4e4468658749a095e0ecf59ab8db2ede02 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Fri, 7 Sep 2012 11:14:01 +0800
Subject: [PATCH 420/641] unicore64: Adjust KSTK_PTREGS_GAP value

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/include/asm/processor.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/unicore64/include/asm/processor.h b/arch/unicore64/include/asm/processor.h
index 8800704..79d226d 100644
--- a/arch/unicore64/include/asm/processor.h
+++ b/arch/unicore64/include/asm/processor.h
@@ -52,7 +52,7 @@ extern int kernel_thread(int (*fn)(void *), void *arg, unsigned long flags);
 extern unsigned long get_wchan(struct task_struct *p);
 
 /* Return some info about the user process TASK. */
-#define KSTK_PTREGS_GAP  8 /* FIXME */
+#define KSTK_PTREGS_GAP		L1_CACHE_BYTES /* Only one word for canary */
 #define task_pt_regs(p)	((struct pt_regs *) \
 		(task_stack_page(p) + THREAD_SIZE - KSTK_PTREGS_GAP) - 1)
 /* Aliases for pc and sp (used in fs/proc/array.c) */
-- 
1.7.9.5

