From e01d0115ca8c85d56d10b679930238c1cd544e0d Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Fri, 7 Sep 2012 14:09:19 +0800
Subject: [PATCH 423/641] unicore64: Remove unused prepare_to_copy

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/include/asm/processor.h |    2 --
 1 file changed, 2 deletions(-)

diff --git a/arch/unicore64/include/asm/processor.h b/arch/unicore64/include/asm/processor.h
index 68f72b7..5684628 100644
--- a/arch/unicore64/include/asm/processor.h
+++ b/arch/unicore64/include/asm/processor.h
@@ -40,8 +40,6 @@ struct thread_struct {
 
 #define thread_saved_pc(tsk)	((tsk)->thread.cpu_context.r30)
 
-/* ____epip: prepare_to_copy will be removed when tip tree merged */
-#define prepare_to_copy(tsk)	do { } while (0)
 #define release_thread(tsk)	do { } while (0)
 
 #define cpu_relax()		barrier()
-- 
1.7.9.5

