From 7b3dbe1d3e7f864d9b7910de1f43064a843840cf Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Fri, 7 Sep 2012 14:06:34 +0800
Subject: [PATCH 424/641] unicore64: Adjust format of KSTK_EIP and KSTK_ESP

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/include/asm/processor.h |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/unicore64/include/asm/processor.h b/arch/unicore64/include/asm/processor.h
index 5684628..61de863 100644
--- a/arch/unicore64/include/asm/processor.h
+++ b/arch/unicore64/include/asm/processor.h
@@ -55,8 +55,8 @@ extern unsigned long get_wchan(struct task_struct *p);
 				(task_stack_page(tsk) + THREAD_SIZE	\
 				- KSTK_PTREGS_GAP - sizeof(struct pt_regs)))
 /* Aliases for pc and sp (used in fs/proc/array.c) */
-#define KSTK_EIP(tsk)  (task_pt_regs(tsk)->UC64_R31)
-#define KSTK_ESP(tsk)  (task_pt_regs(tsk)->UC64_R29)
+#define KSTK_EIP(tsk)		(task_pt_regs(tsk)->UC64_R31)
+#define KSTK_ESP(tsk)		(task_pt_regs(tsk)->UC64_R29)
 
 #endif /* __KERNEL__ */
 
-- 
1.7.9.5

