From 2f58023edd9d5d0a2f2f1b5e4c6e2dd424a7b687 Mon Sep 17 00:00:00 2001
From: Chang Huaixin <changhuaixin@mprc.pku.edu.cn>
Date: Fri, 12 Oct 2012 09:30:02 +0800
Subject: [PATCH 435/641] UniCore64: Add TIF_POLLING_NRFLAG

Signed-off-by: Chang Huaixin <changhuaixin@mprc.pku.edu.cn>
---
 arch/unicore64/include/asm/thread_info.h |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/unicore64/include/asm/thread_info.h b/arch/unicore64/include/asm/thread_info.h
index ef01c59..0550b3a 100644
--- a/arch/unicore64/include/asm/thread_info.h
+++ b/arch/unicore64/include/asm/thread_info.h
@@ -63,6 +63,8 @@ static inline struct thread_info *current_thread_info(void)
 #define TIF_RESTORE_SIGMASK	3 /* restore signal mask in do_signal */
 #define TIF_MEMDIE		4 /* is terminating due to OOM killer */
 #define TIF_NOTIFY_RESUME	5 /* callback before returning to user */
+#define TIF_POLLING_NRFLAG	6 /* true if poll_idle() is polling
+				 TIF_NEED_RESCHED */
 
 #endif /* __KERNEL__ */
 #endif /* __UNICORE64_ASM_THREAD_INFO_H__ */
-- 
1.7.9.5

