From 7ddbe407e4f828748fad0188e67d01e9ec1931cd Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Fri, 26 Oct 2012 15:32:52 +0800
Subject: [PATCH 441/641] unicore64: Add pmd_populate

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/mm/pgtable.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/arch/unicore64/mm/pgtable.c b/arch/unicore64/mm/pgtable.c
index 6c2958e..7e6a312 100644
--- a/arch/unicore64/mm/pgtable.c
+++ b/arch/unicore64/mm/pgtable.c
@@ -117,8 +117,7 @@ void pud_populate(struct mm_struct *mm, pud_t *pudp, pmd_t *pmd)
  */
 void pmd_populate(struct mm_struct *mm, pmd_t *pmd, pgtable_t page)
 {
-	/* FIXME */
-	BUG();
+	set_pmd(pmd, __pmd(page_to_phys(page) | UC64_PMD_EXIST));
 }
 
 /**
-- 
1.7.9.5

