From 4aded7e62e9340ad0677b738bb586c0a59086d61 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Fri, 26 Oct 2012 16:14:06 +0800
Subject: [PATCH 445/641] unicore64: Add pte_offset_kernel

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/include/asm/pgtable.h |    7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/arch/unicore64/include/asm/pgtable.h b/arch/unicore64/include/asm/pgtable.h
index c46f8ef..c9ed790 100644
--- a/arch/unicore64/include/asm/pgtable.h
+++ b/arch/unicore64/include/asm/pgtable.h
@@ -37,12 +37,7 @@ extern pgd_t swapper_pg_dir[PTRS_PER_PGD];
 					pmd_index(addr))
 
 #define pte_index(addr)		(((addr) >> PAGE_SHIFT) & (PTRS_PER_PTE - 1))
-static inline pte_t *pte_offset_kernel(pmd_t *pmd, unsigned long address)
-{
-	/* FIXME */
-	BUG();
-	return 0;
-}
+#define pte_offset_kernel(pmdp, addr)	__va((pte_t *)(pmd_val(*(pmdp)) & PAGE_MASK) + pte_index(addr))
 #define pte_offset_map(dir, addr)	pte_offset_kernel((dir), (addr))
 #define pte_unmap(pte)			do { } while (0)
 #define pte_modify(pte, newprot)	__pte({BUG(); 0; })
-- 
1.7.9.5

