From 205da9b10cf20fa7a8db5324e3ab95cec0f9957a Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Fri, 26 Oct 2012 16:20:35 +0800
Subject: [PATCH 447/641] unicore64: Add set_pte

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/mm/pgtable.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/arch/unicore64/mm/pgtable.c b/arch/unicore64/mm/pgtable.c
index 081052e..c917614 100644
--- a/arch/unicore64/mm/pgtable.c
+++ b/arch/unicore64/mm/pgtable.c
@@ -99,8 +99,7 @@ pte_t *pte_alloc_one_kernel(struct mm_struct *mm, unsigned long address)
  */
 void set_pte(pte_t *ptep, pte_t pte)
 {
-	/* FIXME */
-	BUG();
+	*(ptep) = pte;
 }
 
 /**
-- 
1.7.9.5

