From a2c0512c7028ae5b8f3186ce67abc7038aab8043 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Fri, 26 Oct 2012 16:21:40 +0800
Subject: [PATCH 448/641] unicore64: Make update_mmu_cache noop, fixme lator

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/mm/mmu.c |    1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/unicore64/mm/mmu.c b/arch/unicore64/mm/mmu.c
index 1656141..1e144d6 100644
--- a/arch/unicore64/mm/mmu.c
+++ b/arch/unicore64/mm/mmu.c
@@ -77,5 +77,4 @@ void update_mmu_cache(struct vm_area_struct *vma, unsigned long addr,
 	pte_t *ptep)
 {
 	/* FIXME */
-	BUG();
 }
-- 
1.7.9.5

