From 09eabc4cc222ac038bb80778898c90848d5e4cb1 Mon Sep 17 00:00:00 2001
From: Liang Rongrong <liangrongrong@mprc.pku.edu.cn>
Date: Thu, 23 Aug 2012 11:47:15 +0800
Subject: [PATCH 449/641] UniCore64: Add __pte_free_tlb and __pmd_free_tlb

Signed-off-by: Liang Rongrong <liangrongrong@mprc.pku.edu.cn>
---
 arch/unicore64/include/asm/pgalloc.h |   12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/arch/unicore64/include/asm/pgalloc.h b/arch/unicore64/include/asm/pgalloc.h
index a7dd5f1..9014f6a 100644
--- a/arch/unicore64/include/asm/pgalloc.h
+++ b/arch/unicore64/include/asm/pgalloc.h
@@ -18,7 +18,15 @@ extern void pmd_free(struct mm_struct *, pmd_t *);
 
 extern void pud_populate(struct mm_struct *, pud_t *, pmd_t *);
 
-#define __pte_free_tlb(tlb, ptep, addr)	BUG()
-#define __pmd_free_tlb(tlb, pmdp, addr)	BUG()
+#define __pte_free_tlb(tlb, ptep, addr)				\
+	do {							\
+		pgtable_page_dtor(ptep);			\
+		tlb_remove_page((tlb), (ptep));			\
+	} while (0)
+#define __pmd_free_tlb(tlb, pmdp, addr)				\
+	do {							\
+		pgtable_page_dtor(pmdp);			\
+		tlb_remove_page((tlb), (pmdp));			\
+	} while (0)
 
 #endif /* __UNICORE64_ASM_PGALLOC_H__ */
-- 
1.7.9.5

