From 8b0cc4e67e1e016b48c3f889adb3d8b8ecca6a78 Mon Sep 17 00:00:00 2001
From: Liang Rongrong <liangrongrong@mprc.pku.edu.cn>
Date: Fri, 14 Sep 2012 16:41:28 +0800
Subject: [PATCH 452/641] UniCore64: make the mem_map init earlier

Signed-off-by: Liang Rongrong <liangrongrong@mprc.pku.edu.cn>
---
 arch/unicore64/mm/init.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/unicore64/mm/init.c b/arch/unicore64/mm/init.c
index d498917..25311f1 100644
--- a/arch/unicore64/mm/init.c
+++ b/arch/unicore64/mm/init.c
@@ -102,6 +102,6 @@ void __init setup_arch_memory(void)
 {
 	memblock_init();
 	max_pfn_init();
-	paging_init();
 	zone_sizes_init();
+	paging_init();
 }
-- 
1.7.9.5

