From 74e25dd3b4f77fe5aa9807290a8959f68eb328ae Mon Sep 17 00:00:00 2001
From: Sun Zhichuang <sunzhichuang@mprc.pku.edu.cn>
Date: Fri, 28 Sep 2012 14:40:55 +0800
Subject: [PATCH 455/641] Unicore64 : add __va to calculate vaddr of pmd

---
 arch/unicore64/mm/mmu.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/unicore64/mm/mmu.c b/arch/unicore64/mm/mmu.c
index 1e144d6..c661bc0 100644
--- a/arch/unicore64/mm/mmu.c
+++ b/arch/unicore64/mm/mmu.c
@@ -25,7 +25,7 @@ static void __init uc64_create_direct_mapping(phys_addr_t start,
 	/* the first pud has been written in head.S */
 	prot_pmd = __pgprot(UC64_PMD_TYPE_CACHE | UC64_PMD_EXIST
 			| UC64_PMD_RWX | UC64_PMD_SPAGE);
-	pmd = (pmd_t *)(phys_pud) + pmd_index((unsigned long)__va(start));
+	pmd = (pmd_t *)__va(phys_pud) + pmd_index((unsigned long)__va(start));
 
 	for (phys = start; phys < (start + length); phys += UC64_PMD_SIZE) {
 		set_pmd(pmd, __pmd(phys | pgprot_val(prot_pmd)));
-- 
1.7.9.5

