From 42d9a180c1c8ca129ebe23d10e74ab7528acc154 Mon Sep 17 00:00:00 2001
From: Liang Rongrong <liangrongrong@mprc.pku.edu.cn>
Date: Mon, 19 Nov 2012 15:01:33 +0800
Subject: [PATCH 469/641] UniCore64: Reserve UC64_PM_PGTABLE_PUD_DM00 page

Signed-off-by: Liu Guoli <liuguoli@mprc.pku.edu.cn>
---
 arch/unicore64/mm/init.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/unicore64/mm/init.c b/arch/unicore64/mm/init.c
index 4910156..b96243f 100644
--- a/arch/unicore64/mm/init.c
+++ b/arch/unicore64/mm/init.c
@@ -73,7 +73,7 @@ static void __init memblock_init(void)
 	memblock_reserve(__pa(_text), _end - _text);
 
 	/* Reserve the page tables. */
-	memblock_reserve(__pa(swapper_pg_dir), PAGE_SIZE);
+	memblock_reserve(__pa(swapper_pg_dir), 2 * PAGE_SIZE);
 
 #ifdef CONFIG_BLK_DEV_INITRD
 	if (UC64_PM_INITRD_SIZE) {
-- 
1.7.9.5

