From 2f6eb6cb0c638e1a5847b9ba90ecb32b36b67161 Mon Sep 17 00:00:00 2001
From: Liu Guoli <liuguoli@mprc.pku.edu.cn>
Date: Mon, 19 Nov 2012 15:39:29 +0800
Subject: [PATCH 471/641] UniCore64: Add __vec_jepriv

Signed-off-by: Liu Guoli <liuguoli@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/entry.S |   45 ++++++++++++++++++++++++++++++++++++++++-
 1 file changed, 44 insertions(+), 1 deletion(-)

diff --git a/arch/unicore64/kernel/entry.S b/arch/unicore64/kernel/entry.S
index 71e8b5f..c32864b 100644
--- a/arch/unicore64/kernel/entry.S
+++ b/arch/unicore64/kernel/entry.S
@@ -86,6 +86,49 @@ ENTRY(__vec_invalid)
 	__halt					@ no return for invalid vec
 ENDPROC(__vec_invalid)
 
+ENTRY(__vec_jepriv)
+	std.wu		sp, [sp-], #8
+	std.wu		lr, [sp-], #8
+	__context_save
+
+	/*
+	 * Get the system call number.
+	 */
+	movc	r16, CP0_TRAPADDR, #1		@ saved in epc reg
+	dsub r16, r16, #4
+	ldw r16, [r16+], #0
+
+	__irq_enable
+
+	ldd	r17, =sys_call_table		@ load syscall table pointer
+
+	dmovl	r18, #0xffffff		@ mask off SWI op-code
+	dand	r16, r16, r18		@ mask off SWI op-code
+
+	dlsl r16, r16, #3
+	ldd	r18, [r17+], r16	@ call sys_* routine
+	std.w	r5, [sp-], #8		@ push fifth and sixth args
+	std.w	r4, [sp-], #8
+	call.r r18
+	dadd	sp, sp, #16
+	std.w	r0, [sp+], #0
+
+ret_syscall:
+	__irq_disable
+	__thread_info	r17
+	ldw		r18, [r17+], #THREAD_INFO_FLAGS
+	and		r0, r18, #(1 << TIF_NEED_RESCHED)
+	cmpsub.a	r0, #0
+	adr		lr, ret_syscall
+	bne		schedule
+	cmpsub.a	r18, #0
+	bne		__vec_invalid		@ print error information
+	__context_restore
+	ldd.wu		lr, [sp]+, #8
+	ldd.wu		sp, [sp]+, #8
+	eret
+ENDPROC(__vec_jepriv)
+
 ENTRY(__vec_itrap)
 	__push		sp				@ push r29(sp)
 	__push		lr				@ push r30(lr)
@@ -170,7 +213,7 @@ ENDPROC(__vec_int_itimer)
 ENTRY(__vectors_table)
 	call		__vec_invalid			@ 0x00: RESET
 	call		__vec_invalid			@ 0x04: EEXTN
-	call		__vec_invalid			@ 0x08: ESWI
+	b		__vec_jepriv			@ 0x08: ESWI
 	b		__vec_itrap			@ 0x0c: ITRAP
 	b		__vec_dtrap			@ 0x10: DTRAP
 	call		__vec_invalid			@ 0x14: FPU_EXC
-- 
1.7.9.5

