From 6ee3bd6562e215948d0829d902fe689058e92fd2 Mon Sep 17 00:00:00 2001
From: Liang Rongrong <liangrongrong@mprc.pku.edu.cn>
Date: Mon, 19 Nov 2012 16:50:15 +0800
Subject: [PATCH 475/641] UniCore64: Fix the BUG in copy_thread

Signed-off-by: Liang Rongrong <liangrongrong@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/process.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/unicore64/kernel/process.c b/arch/unicore64/kernel/process.c
index b85ad2f..c84e994 100644
--- a/arch/unicore64/kernel/process.c
+++ b/arch/unicore64/kernel/process.c
@@ -25,8 +25,8 @@ int copy_thread(unsigned long clone_flags, unsigned long stack_start,
 
 	if (unlikely(!regs)) {
 		p->thread.cpu_context.r30 = (unsigned long)ret_from_kthread;
-		p->thread.cpu_context.r17 = stack_start; /* arg for fn(arg) */
-		p->thread.cpu_context.r18 = stk_sz; /* fn for fn(arg) */
+		p->thread.cpu_context.r18 = stack_start; /* fn for fn(arg) */
+		p->thread.cpu_context.r17 = stk_sz; /* arg for fn(arg) */
 		memset(childregs, 0, sizeof(struct pt_regs));
 		/* For kernel_thread, useless pt_regs will be kept in stack */
 		childregs->UC64_R29 = (unsigned long)childregs;
-- 
1.7.9.5

