From 93d81674c3fa0435a8b576ec2b49a5739d2d7938 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Tue, 20 Nov 2012 19:52:42 +0800
Subject: [PATCH 478/641] unicore64: Make r16 as temp reg in kernel

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/include/arch/asm-common.h |    4 ----
 1 file changed, 4 deletions(-)

diff --git a/arch/unicore64/include/arch/asm-common.h b/arch/unicore64/include/arch/asm-common.h
index 1b1d7d9..b5f77057 100644
--- a/arch/unicore64/include/arch/asm-common.h
+++ b/arch/unicore64/include/arch/asm-common.h
@@ -26,19 +26,15 @@ __ASMMACRO_WRAP(.macro	__halt;
 		.endm)
 
 __ASMMACRO_WRAP(.macro	__irq_disable;
-			movc	p0.c12, r16, #0;
 			dmov	r16, asr;
 			dor	r16, r16, #ASR_INTR_SELECT;
 			dmov	asr, r16;
-			movc	r16, p0.c12, #0;
 		.endm)
 
 __ASMMACRO_WRAP(.macro	__irq_enable;
-			movc	p0.c12, r16, #0;
 			dmov	r16, asr;
 			dand	r16, r16, #(~ASR_INTR_SELECT);
 			dmov	asr, r16;
-			movc	r16, p0.c12, #0;
 		.endm)
 
 #define __halt()			__asm__("__halt")
-- 
1.7.9.5

