From ddd41c75d716227481faf9fc12068c641b0cb008 Mon Sep 17 00:00:00 2001
From: Liu Guoli <liuguoli@mprc.pku.edu.cn>
Date: Tue, 20 Nov 2012 17:43:35 +0800
Subject: [PATCH 480/641] UniCore64: Implement cpu_idle()

Signed-off-by: Liu Guoli <liuguoli@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/process.c |    9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/arch/unicore64/kernel/process.c b/arch/unicore64/kernel/process.c
index c84e994..68338b4 100644
--- a/arch/unicore64/kernel/process.c
+++ b/arch/unicore64/kernel/process.c
@@ -112,6 +112,11 @@ unsigned long get_wchan(struct task_struct *p)
  */
 void cpu_idle(void)
 {
-	/* FIXME */
-	pr_info("cpu_idle is doing nothing now.\n");
+	/* endless idle loop with no priority at all */
+	while (1) {
+		/* FIXME: here we do some critical things */
+		preempt_enable_no_resched();
+		schedule();
+		preempt_disable();
+	}
 }
-- 
1.7.9.5

