From a4e3fbea3b061a69fcfc86eda69fbd9645ec9d3b Mon Sep 17 00:00:00 2001
From: Liu Guoli <liuguoli@mprc.pku.edu.cn>
Date: Fri, 30 Nov 2012 11:27:56 +0800
Subject: [PATCH 487/641] UniCore64: Fix bug in puv3_intc_handler

Signed-off-by: Liu Guoli <liuguoli@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/puv3-intc.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/unicore64/kernel/puv3-intc.c b/arch/unicore64/kernel/puv3-intc.c
index c083444..91ed244 100644
--- a/arch/unicore64/kernel/puv3-intc.c
+++ b/arch/unicore64/kernel/puv3-intc.c
@@ -95,7 +95,7 @@ void puv3_intc_handler(void)
 		if (irqs == 0)
 			return;
 
-		irq = ffs(irqs);
+		irq = ffs(irqs) - 1;
 		irq_enter();
 		generic_handle_irq(irq);
 		irq_exit();
-- 
1.7.9.5

