From 59d70d535f776bfd6b9e81701be160b48787beaf Mon Sep 17 00:00:00 2001
From: Chang Huaixin <changhuaixin@mprc.pku.edu.cn>
Date: Sat, 15 Dec 2012 15:26:07 +0800
Subject: [PATCH 492/641] unicore64: Add arch_read_lock and arch_read_unlock

Signed-off-by: Chang Huaixin <changhuaixin@mprc.pku.edu.cn>
---
 arch/unicore64/include/asm/spinlock.h |   38 ++++++++++++++++++++++++---------
 1 file changed, 28 insertions(+), 10 deletions(-)

diff --git a/arch/unicore64/include/asm/spinlock.h b/arch/unicore64/include/asm/spinlock.h
index a52564c..9dadcd1 100644
--- a/arch/unicore64/include/asm/spinlock.h
+++ b/arch/unicore64/include/asm/spinlock.h
@@ -57,20 +57,38 @@ static inline int arch_spin_trylock(arch_spinlock_t *lock)
 
 static inline void arch_read_lock(arch_rwlock_t *rw)
 {
-	/* FIXME */
-	__asm__(
-		"dmovl r0, 0xdead0004\n\t"
-		"call uc64_debug_putx\n\t"
-	);
+	u32 tmp;
+
+	__asm__ __volatile__(
+		"1:	llw		%0, [%1+], #0\n"
+		"	cmpsub.a	%0, #0\n"
+		"	bfs		1b\n"
+		"	add		%0, %0, #1\n"
+		"	scw		%0, [%1+], #0\n"
+		"	cmpsub.a	%0, #0\n"
+		"	beq		1b"
+		: "=&r" (tmp)
+		: "r" (&rw->lock)
+		: "cc", "memory");
+
+	smp_mb();
 }
 
 static inline void arch_read_unlock(arch_rwlock_t *rw)
 {
-	/* FIXME */
-	__asm__(
-		"dmovl r0, 0xdead0005\n\t"
-		"call uc64_debug_putx\n\t"
-	);
+	u32 tmp;
+
+	smp_mb();
+
+	__asm__ __volatile__(
+		"1:	llw		%0, [%1+], #0\n"
+		"	sub		%0, %0, #1\n"
+		"	scw		%0,	[%1+], #0\n"
+		"	cmpsub.a	%0, #0\n"
+		"	beq		1b"
+		: "=&r" (tmp)
+		: "r" (&rw->lock)
+		: "cc", "memory");
 }
 
 static inline int arch_read_trylock(arch_rwlock_t *rw)
-- 
1.7.9.5

