From f89cac1f52bf0921a8f363cb6ed475bc6e46879c Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Wed, 19 Dec 2012 12:42:50 +0800
Subject: [PATCH 500/641] unicore64: Add sync asmmacro

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/include/arch/asm-common.h |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/unicore64/include/arch/asm-common.h b/arch/unicore64/include/arch/asm-common.h
index 65f7bc6..094c4f0 100644
--- a/arch/unicore64/include/arch/asm-common.h
+++ b/arch/unicore64/include/arch/asm-common.h
@@ -25,6 +25,10 @@ __ASMMACRO_WRAP(.macro	__halt;
 			halt;
 		.endm)
 
+__ASMMACRO_WRAP(.macro	__sync;
+			sync;
+		.endm)
+
 __ASMMACRO_WRAP(.macro	__irq_disable;
 			dmov	r16, asr;
 			dor	r16, r16, #ASR_INTR_SELECT;
@@ -38,6 +42,7 @@ __ASMMACRO_WRAP(.macro	__irq_enable;
 		.endm)
 
 #define __halt()			__asm__ __volatile__("__halt")
+#define __sync()			__asm__ __volatile__("__sync")
 
 /*
  * We use "____" prefix(4 '_') only for debug:
-- 
1.7.9.5

