From 90bb613fb95f32cdb801a001b64ee27193f2046a Mon Sep 17 00:00:00 2001
From: Chang Huaixin <changhuaixin@mprc.pku.edu.cn>
Date: Tue, 18 Dec 2012 17:23:21 +0800
Subject: [PATCH 504/641] unicore64: Modify arch_spin_trylock

Signed-off-by: Chang Huaixin <changhuaixin@mprc.pku.edu.cn>
---
 arch/unicore64/include/asm/spinlock.h |   10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/arch/unicore64/include/asm/spinlock.h b/arch/unicore64/include/asm/spinlock.h
index a600267..5e81072 100644
--- a/arch/unicore64/include/asm/spinlock.h
+++ b/arch/unicore64/include/asm/spinlock.h
@@ -53,21 +53,19 @@ static inline int arch_spin_trylock(arch_spinlock_t *lock)
 	__asm__ __volatile__(
 		"	llw		%0, [%1+], #0\n"
 		"	cmpsub.a	%0, #0\n"
+		"	cmovne		%0, #0\n"
 		"	bne		1f\n"
 		"	mov		%0, %2\n"
 		"	scw		%0, [%1+], #0\n"
-		"	sub		%0, %0, #1\n"
 		"1:"
 		: "=&r" (tmp)
 		: "r" (&lock->lock), "r" (LOCK_TOKEN)
 		: "cc", "memory");
 
-	if (tmp == 0) {
+	if (tmp)
 		smp_mb();
-		return 1;
-	} else {
-		return 0;
-	}
+
+	return tmp;
 }
 
 static inline void arch_read_lock(arch_rwlock_t *rw)
-- 
1.7.9.5

