From 111fd381a8b6b0231996df9036fe6c6efa919c04 Mon Sep 17 00:00:00 2001
From: Liang Rongrong <liangrongrong@mprc.pku.edu.cn>
Date: Wed, 2 Jan 2013 16:33:25 +0800
Subject: [PATCH 526/641] UniCore64: Fix the bug in __context_restore

Signed-off-by: Liang Rongrong <liangrongrong@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/entry.S |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/unicore64/kernel/entry.S b/arch/unicore64/kernel/entry.S
index 9158168..2d77824 100644
--- a/arch/unicore64/kernel/entry.S
+++ b/arch/unicore64/kernel/entry.S
@@ -53,9 +53,9 @@
 	dmov		bfr, lr
 	__pop		lr				@ pop bsr
 	dmov		bsr, lr
+	dadd		sp, sp, #16			@ Reserved 2 dwords
 	__pop		lr				@ pop swr #0
 	movc		CP0_SYSU, lr, #0
-	dadd		sp, sp, #16			@ Reserved 2 dwords
 	dadd		sp, sp, #8			@ pop return value
 	__pop		lr				@ pop r31(pc)
 	movc		CP0_TRAPADDR, lr, #1		@ saved in epc reg
-- 
1.7.9.5

