From 50ba32e5c8b8bbc6fa08b616bf85422772bc165b Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Fri, 4 Jan 2013 17:12:49 +0800
Subject: [PATCH 527/641] unicore64: Remove -pg option for puv4-intc.c

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/Makefile |    1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/unicore64/kernel/Makefile b/arch/unicore64/kernel/Makefile
index ee7dd38..caf9cc8 100644
--- a/arch/unicore64/kernel/Makefile
+++ b/arch/unicore64/kernel/Makefile
@@ -7,6 +7,7 @@ ccflags-$(CONFIG_DEBUG_MCOUNT)	:= -pg
 CFLAGS_REMOVE_early_printk.o	:= -pg
 CFLAGS_REMOVE_time.o		:= -pg
 CFLAGS_REMOVE_process.o		:= -pg
+CFLAGS_REMOVE_puv4-intc.o	:= -pg
 
 obj-y				:= head.o entry.o switch_to.o
 obj-y				+= irq.o
-- 
1.7.9.5

