From 72eb567db8ea8ba2de3d10fc7c70773110a109fb Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Wed, 9 Jan 2013 15:18:29 +0800
Subject: [PATCH 529/641] unicore64: Not restore ksp when exit from exceptions

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/entry.S |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/unicore64/kernel/entry.S b/arch/unicore64/kernel/entry.S
index 3b79c54..c0870f9 100644
--- a/arch/unicore64/kernel/entry.S
+++ b/arch/unicore64/kernel/entry.S
@@ -66,7 +66,7 @@
 
 	@ step 4: pop kernel-mode sp and lr from pt_regs
 	__pop		lr
-	__pop		sp
+	dadd            sp, sp, #8                      @ pop ksp
 	.endm
 
 /**
-- 
1.7.9.5

