From e8888340df1ac8252eba5b19e21cbd7c1fcbbbb1 Mon Sep 17 00:00:00 2001
From: Liu Guoli <liuguoli@mprc.pku.edu.cn>
Date: Wed, 9 Jan 2013 14:15:17 +0800
Subject: [PATCH 530/641] UniCore64: Fix BUG in copy_thread

Signed-off-by: Liu Guoli <liuguoli@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/process.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/unicore64/kernel/process.c b/arch/unicore64/kernel/process.c
index e790d93..c3033f2 100644
--- a/arch/unicore64/kernel/process.c
+++ b/arch/unicore64/kernel/process.c
@@ -37,7 +37,7 @@ int copy_thread(unsigned long clone_flags, unsigned long stack_start,
 	}
 
 	/* Kernel-mode SP should be ALWAYS kept for re-entrance */
-	childregs->UC64_KSP = (unsigned long)childregs;
+	childregs->UC64_KSP = (unsigned long)childregs + sizeof(struct pt_regs);
 
 	return 0;
 }
-- 
1.7.9.5

