From fd697857d311e3286dc565d342d735c8ad8e4599 Mon Sep 17 00:00:00 2001
From: Chang Huaixin <changhuaixin@mprc.pku.edu.cn>
Date: Fri, 28 Dec 2012 15:26:27 +0800
Subject: [PATCH 533/641] unicore64: Add xchg

Signed-off-by: Chang Huaixin <changhuaixin@mprc.pku.edu.cn>
---
 arch/unicore64/include/asm/cmpxchg.h |   37 +++++++++++++++++++++++++++++-----
 1 file changed, 32 insertions(+), 5 deletions(-)

diff --git a/arch/unicore64/include/asm/cmpxchg.h b/arch/unicore64/include/asm/cmpxchg.h
index 98c3f4f..2077e9f 100644
--- a/arch/unicore64/include/asm/cmpxchg.h
+++ b/arch/unicore64/include/asm/cmpxchg.h
@@ -65,11 +65,38 @@ static inline unsigned long __cmpxchg(volatile void *ptr, unsigned long old,
 					  (unsigned long)(n),		\
 					  sizeof(*(ptr))))
 
-#define xchg(ptr, x)				\
-	({					\
-		uc64_debug_putx(0xdead0022);	\
-		(*ptr);				\
-	})
+extern void __xchg_called_with_bad_pointer(void);
+
+static inline unsigned long __xchg(unsigned long x, volatile void *ptr,
+				int size)
+{
+	unsigned long ret;
+
+	switch (size) {
+	case 4:
+		__asm__ __volatile__(
+			"swapw	%0, [%1], %2"
+			: "=&r" (ret)
+			: "r" (ptr), "r" (x)
+			: "cc", "memory");
+		break;
+	case 8:
+		__asm__ __volatile__(
+			"swapd	%0, [%1], %2"
+			: "=&r" (ret)
+			: "r" (ptr), "r" (x)
+			: "cc", "memory");
+		break;
+	default:
+		__xchg_called_with_bad_pointer();
+	}
+
+	return ret;
+}
+
+#define xchg(ptr, x) \
+	((__typeof__(*(ptr)))__xchg((unsigned long)(x), (ptr), sizeof(*(ptr))))
+
 #endif /* CONFIG_SMP */
 
 #endif /* __UNICORE64_ASM_CMPXCHG_H__ */
-- 
1.7.9.5

