From e0425f7f54e73267e2301a2a78209f0fd4423516 Mon Sep 17 00:00:00 2001
From: Chang Huaixin <changhuaixin@mprc.pku.edu.cn>
Date: Mon, 19 Nov 2012 17:45:07 +0800
Subject: [PATCH 537/641] unicore64: set cpumask for __itimer_ce

Signed-off-by: Chang Huaixin <changhuaixin@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/time.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/unicore64/kernel/time.c b/arch/unicore64/kernel/time.c
index 9c27293..0bc2eca4 100644
--- a/arch/unicore64/kernel/time.c
+++ b/arch/unicore64/kernel/time.c
@@ -137,6 +137,8 @@ void __init time_init(void)
 	__itimer_ce.min_delta_ns =
 		clockevent_delta2ns(MIN_COUNTER_DELTA, &__itimer_ce);
 
+	__itimer_ce.cpumask = cpumask_of(smp_processor_id());
+
 	clocksource_register_hz(&__itimer_cs, CLOCK_TICK_RATE);
 	clockevents_register_device(&__itimer_ce);
 }
-- 
1.7.9.5

