From 0e3bc8b9d3909067215dc29163ea46597740f1cb Mon Sep 17 00:00:00 2001
From: Chang Huaixin <changhuaixin@mprc.pku.edu.cn>
Date: Wed, 16 Jan 2013 11:03:17 +0800
Subject: [PATCH 538/641] unicore64: Set cpu possible map

Signed-off-by: Chang Huaixin <changhuaixin@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/smp.c |   10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/unicore64/kernel/smp.c b/arch/unicore64/kernel/smp.c
index a0e78bd..747f36a 100644
--- a/arch/unicore64/kernel/smp.c
+++ b/arch/unicore64/kernel/smp.c
@@ -1,6 +1,16 @@
 #include <linux/init.h>
 #include <linux/smp.h>
 
+unsigned int num_processors = 2;
+
+void __init smp_setup_processor_id(void)
+{
+	int i;
+
+	for (i = 0; i < num_processors; i++)
+		set_cpu_possible(i, true);
+}
+
 void __init smp_prepare_cpus(unsigned int max_cpus)
 {
 	/* FIXME */
-- 
1.7.9.5

