From 4ca808f3faa3774ca30247663744ef36abba7002 Mon Sep 17 00:00:00 2001
From: Chang Huaixin <changhuaixin@mprc.pku.edu.cn>
Date: Mon, 28 Jan 2013 16:05:03 +0800
Subject: [PATCH 541/641] unicore64: Set cpu present map

Signed-off-by: Chang Huaixin <changhuaixin@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/smp.c |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/arch/unicore64/kernel/smp.c b/arch/unicore64/kernel/smp.c
index aa2316c..685fdec 100644
--- a/arch/unicore64/kernel/smp.c
+++ b/arch/unicore64/kernel/smp.c
@@ -13,8 +13,10 @@ void __init smp_setup_processor_id(void)
 
 void __init smp_prepare_cpus(unsigned int max_cpus)
 {
-	/* FIXME */
-	BUG();
+	int cpu;
+
+	for_each_possible_cpu(cpu)
+		set_cpu_present(cpu, true);
 }
 
 void __devinit smp_prepare_boot_cpu(void)
-- 
1.7.9.5

