From 3a8e6a34dfb9623f467b64dbf86c627700b09a9d Mon Sep 17 00:00:00 2001
From: Chang Huaixin <changhuaixin@mprc.pku.edu.cn>
Date: Mon, 28 Jan 2013 16:15:27 +0800
Subject: [PATCH 542/641] unicore64: Use generic idle thread

Signed-off-by: Chang Huaixin <changhuaixin@mprc.pku.edu.cn>
---
 arch/unicore64/Kconfig |    1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/unicore64/Kconfig b/arch/unicore64/Kconfig
index 4f048db9b..d34544c 100644
--- a/arch/unicore64/Kconfig
+++ b/arch/unicore64/Kconfig
@@ -10,6 +10,7 @@ config UNICORE64
 	select GENERIC_IRQ_SHOW
 	select GENERIC_KERNEL_THREAD
 	select GENERIC_KERNEL_EXECVE
+	select GENERIC_SMP_IDLE_THREAD
 	select OF
 	select OF_EARLY_FLATTREE
 	select NO_BOOTMEM
-- 
1.7.9.5

