From 9ef2fa5523f23b06cbaa7873a05ad692e0bc3db5 Mon Sep 17 00:00:00 2001
From: Chang Huaixin <changhuaixin@mprc.pku.edu.cn>
Date: Mon, 27 May 2013 16:28:04 +0800
Subject: [PATCH 546/641] unicore64: Add AP initialization mainframe

Signed-off-by: Chang Huaixin <changhuaixin@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/smp.c |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/arch/unicore64/kernel/smp.c b/arch/unicore64/kernel/smp.c
index 8cfcf09..45e2a07 100644
--- a/arch/unicore64/kernel/smp.c
+++ b/arch/unicore64/kernel/smp.c
@@ -123,8 +123,11 @@ int __cpuinit __cpu_up(unsigned int cpu, struct task_struct *idle)
  */
 void __init secondary_start_kernel(void)
 {
-	/* FIXME */
-	BUG();
+	preempt_disable();
+
+	smp_secondary_alive = 1;
+
+	cpu_idle();
 }
 
 void smp_send_stop(void)
-- 
1.7.9.5

