From 81efd89467aba21b02df673135f37fff64e220d2 Mon Sep 17 00:00:00 2001
From: Chang Huaixin <changhuaixin@mprc.pku.edu.cn>
Date: Tue, 28 May 2013 15:42:25 +0800
Subject: [PATCH 556/641] unicore64: Increase mm_count for init_mm

Signed-off-by: Chang Huaixin <changhuaixin@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/smp.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/unicore64/kernel/smp.c b/arch/unicore64/kernel/smp.c
index 01634bb..30cbc5e 100644
--- a/arch/unicore64/kernel/smp.c
+++ b/arch/unicore64/kernel/smp.c
@@ -220,6 +220,9 @@ int __cpuinit __cpu_up(unsigned int cpu, struct task_struct *idle)
  */
 void __init secondary_start_kernel(void)
 {
+	atomic_inc(&init_mm.mm_count);
+	current->active_mm = &init_mm;
+
 	preempt_disable();
 
 	smp_secondary_alive = 1;
-- 
1.7.9.5

