From f6506060bed4dcdcb2574163aa0c15db10bc13b5 Mon Sep 17 00:00:00 2001
From: Chang Huaixin <changhuaixin@mprc.pku.edu.cn>
Date: Mon, 17 Jun 2013 16:57:52 +0800
Subject: [PATCH 558/641] unicore64: Notify cpu start

Signed-off-by: Chang Huaixin <changhuaixin@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/smp.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/unicore64/kernel/smp.c b/arch/unicore64/kernel/smp.c
index 5f47b85..c3940da 100644
--- a/arch/unicore64/kernel/smp.c
+++ b/arch/unicore64/kernel/smp.c
@@ -227,6 +227,8 @@ void __init secondary_start_kernel(void)
 
 	setup_itimer();
 
+	notify_cpu_starting(cpu);
+
 	smp_secondary_alive = 1;
 
 	cpu_idle();
-- 
1.7.9.5

