From 180006e66160f94494e9d43a5ed93a57f1e962d9 Mon Sep 17 00:00:00 2001
From: Chang Huaixin <changhuaixin@mprc.pku.edu.cn>
Date: Mon, 17 Jun 2013 17:39:26 +0800
Subject: [PATCH 561/641] unicore64: AP calibrate_delay

Signed-off-by: Chang Huaixin <changhuaixin@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/smp.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/unicore64/kernel/smp.c b/arch/unicore64/kernel/smp.c
index 1150c64..f29106f 100644
--- a/arch/unicore64/kernel/smp.c
+++ b/arch/unicore64/kernel/smp.c
@@ -235,6 +235,8 @@ void __init secondary_start_kernel(void)
 
 	local_irq_enable();
 
+	calibrate_delay();
+
 	smp_secondary_alive = 1;
 
 	cpu_idle();
-- 
1.7.9.5

