From c970f35e2bdf9862da7dab3e93d74f9b14987bf2 Mon Sep 17 00:00:00 2001
From: Li Binbin <libinbin@mprc.pku.edu.cn>
Date: Wed, 19 Feb 2014 14:34:32 +0800
Subject: [PATCH 566/641] unicore64: Add Page Alignment

Signed-off-by: Li Binbin <libinbin@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/vmlinux.lds.S |    1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/unicore64/kernel/vmlinux.lds.S b/arch/unicore64/kernel/vmlinux.lds.S
index 4f1c885..686511b 100644
--- a/arch/unicore64/kernel/vmlinux.lds.S
+++ b/arch/unicore64/kernel/vmlinux.lds.S
@@ -24,6 +24,7 @@ SECTIONS
 	INIT_TEXT_SECTION(PAGE_SIZE)
 	INIT_DATA_SECTION(L1_CACHE_BYTES)
 	PERCPU_SECTION(L1_CACHE_BYTES)
+	. = ALIGN(PAGE_SIZE);
 	__init_end = .;
 
 	_stext = .;
-- 
1.7.9.5

