From 39ac93667066a6a06eb360c5279a1b332ca7fe23 Mon Sep 17 00:00:00 2001
From: Chang Huaixin <changhuaixin@mprc.pku.edu.cn>
Date: Mon, 3 Mar 2014 16:39:57 +0800
Subject: [PATCH 572/641] unicore64: Define CSR temporary reg

Signed-off-by: Chang Huaixin <changhuaixin@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/smp.c |   10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/arch/unicore64/kernel/smp.c b/arch/unicore64/kernel/smp.c
index f5ab0bc..331d542 100644
--- a/arch/unicore64/kernel/smp.c
+++ b/arch/unicore64/kernel/smp.c
@@ -165,6 +165,11 @@ void __init smp_cpus_done(unsigned int max_cpus)
 
 unsigned long secondary_stack_start;
 
+/*
+ * CSR is not quite ready. Use a temporary reg 0xff6100000.
+ */
+#define CSR_R0	0xfffffffff6100000
+
 static int __init smp_boot_one_cpu(unsigned int cpu, struct task_struct *idle)
 {
 	unsigned long timeout;
@@ -173,11 +178,8 @@ static int __init smp_boot_one_cpu(unsigned int cpu, struct task_struct *idle)
 
 	wmb();
 
-	/*
-	 * CSU is not quite ready. Use a temporary reg 0xff6100000.
-	 */
 	__asm__ __volatile__(
-		"dmovl	r0, #0xfffffffff6100000\n"
+		"dmovl	r0, #" __stringify(CSR_R0) "\n"
 		"ldd	r1, =secondary_stext\n"
 		"stw	r1, [r0]\n"
 		"movc	p0.c10, r0, #4");
-- 
1.7.9.5

