From 36b6d7a285b015e136dfe8b3085b74be671ee65d Mon Sep 17 00:00:00 2001
From: Chang Huaixin <changhuaixin@mprc.pku.edu.cn>
Date: Mon, 3 Mar 2014 16:40:39 +0800
Subject: [PATCH 573/641] unicore64: Modify secondary_stack_start

Signed-off-by: Chang Huaixin <changhuaixin@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/smp.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/unicore64/kernel/smp.c b/arch/unicore64/kernel/smp.c
index 331d542..8825f34 100644
--- a/arch/unicore64/kernel/smp.c
+++ b/arch/unicore64/kernel/smp.c
@@ -163,7 +163,7 @@ void __init smp_cpus_done(unsigned int max_cpus)
 		(bogosum / (5000/HZ)) % 100);
 }
 
-unsigned long secondary_stack_start;
+volatile unsigned long secondary_stack_start;
 
 /*
  * CSR is not quite ready. Use a temporary reg 0xff6100000.
@@ -212,6 +212,7 @@ int __cpuinit __cpu_up(unsigned int cpu, struct task_struct *idle)
 	 */
 	secondary_stack_start = (unsigned long) task_stack_page(idle)
 			+ THREAD_SIZE - KSTK_PTREGS_GAP;
+	mb();
 
 	/*
 	 * Now bring the CPU into our world.
-- 
1.7.9.5

