From dca50a4dc64782bd4447829dff7927939996f930 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Tue, 4 Mar 2014 02:09:31 +0800
Subject: [PATCH 577/641] UniCore64: Add kernel mode dtrap exit handling

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/entry.S |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/arch/unicore64/kernel/entry.S b/arch/unicore64/kernel/entry.S
index e0c7937..6af0b5b 100644
--- a/arch/unicore64/kernel/entry.S
+++ b/arch/unicore64/kernel/entry.S
@@ -193,7 +193,11 @@ ENTRY(__vec_dtrap)
 
 	__irq_disable
 
-	b		ret_to_user
+	ldd		r17, [sp+], #240	@ get bsr
+	dand		r17, r17, #7
+	cmpsub.a	r17, #1
+	beq		ret_to_user
+
 	__context_restore
 	eret
 ENDPROC(__vec_dtrap)
-- 
1.7.9.5

