From f432367358fdd18d6601f0e44ef1558c415b6e72 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Thu, 27 Mar 2014 19:20:16 +0800
Subject: [PATCH 581/641] UniCore64: modify comment for memblock_reserve

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/mm/init.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/unicore64/mm/init.c b/arch/unicore64/mm/init.c
index a017847..5d8638d 100644
--- a/arch/unicore64/mm/init.c
+++ b/arch/unicore64/mm/init.c
@@ -75,7 +75,7 @@ static void __init memblock_init(void)
 	/* Reserve the kernel text, kernel data and initrd with memblock. */
 	memblock_reserve(__pa(_text), _end - _text);
 
-	/* Reserve the page tables. */
+	/* Reserve zero page, DTB and page tables. */
 	memblock_reserve(__pa(UC64_VM_DTB_START), 8 * PAGE_SIZE);
 
 	memblock_allow_resize();
-- 
1.7.9.5

