From 67f4087c8078eee000605df43ce218d4b42b2de9 Mon Sep 17 00:00:00 2001
From: Li Binbin <libinbin@mprc.pku.edu.cn>
Date: Wed, 16 Apr 2014 23:46:50 +0800
Subject: [PATCH 583/641] unicore64: Fix BUG in do_signal

Signed-off-by: Li Binbin <libinbin@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/signal.c |   27 +++++++++++++++++++++++++--
 1 file changed, 25 insertions(+), 2 deletions(-)

diff --git a/arch/unicore64/kernel/signal.c b/arch/unicore64/kernel/signal.c
index d729a40..e02fdf4 100644
--- a/arch/unicore64/kernel/signal.c
+++ b/arch/unicore64/kernel/signal.c
@@ -110,8 +110,31 @@ static void do_signal(struct pt_regs *regs, int syscall)
 			force_sigsegv(signr, current);
 		else
 			signal_delivered(signr, &info, &ka, regs, false);
-	} else
-		BUG(); /* FIXME: did we come from a system call? */
+	}
+
+	/* Did we come from a system call? */
+	if (syscall) {
+		/* Restart the system call - no handlers present */
+		switch (regs->UC64_R00) {
+		case -ERESTARTNOHAND:
+		case -ERESTARTSYS:
+		case -ERESTARTNOINTR:
+			regs->UC64_R00 = regs->UC64_O00;
+			regs->UC64_R31 -= 4;
+			break;
+
+		case -ERESTART_RESTARTBLOCK:
+			regs->UC64_R00 = __NR_restart_syscall;
+			regs->UC64_R31 -= 4;
+			break;
+		}
+	}
+
+	/*
+	 * If there's no signal to deliver, we just put the saved sigmask
+	 * back.
+	 */
+	restore_saved_sigmask();
 }
 
 asmlinkage void do_notify_resume(unsigned int thread_flags, int syscall)
-- 
1.7.9.5

