From 68af678ca9cf8c30df43803e6a39edef118f2923 Mon Sep 17 00:00:00 2001
From: Li Binbin <libinbin@mprc.pku.edu.cn>
Date: Wed, 16 Apr 2014 23:50:02 +0800
Subject: [PATCH 584/641] unicore64: Fix deaddead info print in entry.S

Signed-off-by: Li Binbin <libinbin@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/entry.S |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/unicore64/kernel/entry.S b/arch/unicore64/kernel/entry.S
index 6af0b5b..1dd5d7d 100644
--- a/arch/unicore64/kernel/entry.S
+++ b/arch/unicore64/kernel/entry.S
@@ -146,7 +146,8 @@ ret_to_user:
 	bne.l		do_notify_resume
 	__thread_info	r17
 	ldw		r18, [r17+], #THREAD_INFO_FLAGS
-	cmpsub.a	r18, #0
+	and.a		r0, r18, #(1 << TIF_NOTIFY_RESUME | 1 << TIF_SIGPENDING)
+	cmpsub.a	r0, #0
 	bne		__vec_invalid		@ print error information
 	__context_restore
 	eret
-- 
1.7.9.5

