From bc98ae4bf0751cf15785cbb98158a1592a2eaa1a Mon Sep 17 00:00:00 2001
From: Liu Guoli <liuguoli@mprc.pku.edu.cn>
Date: Mon, 21 Apr 2014 18:34:23 +0800
Subject: [PATCH 588/641] Debug: add tlb flush after paging init

Signed-off-by: Liu Guoli <liuguoli@mprc.pku.edu.cn>
---
 arch/unicore64/mm/mmu.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/unicore64/mm/mmu.c b/arch/unicore64/mm/mmu.c
index 997f4b6..763a42a 100644
--- a/arch/unicore64/mm/mmu.c
+++ b/arch/unicore64/mm/mmu.c
@@ -3,6 +3,7 @@
 #include <linux/sched.h>
 
 #include <asm/setup_arch.h>
+#include <arch/asm-mmuops.h>
 
 /*
  * empty_zero_page is a special page that is used for
@@ -75,6 +76,8 @@ void __init paging_init(void)
 
 	uc64_create_io_direct_mapping();
 
+	__invalid_tlb();
+
 }
 
 /**
-- 
1.7.9.5

