From 28b16a42637990d2798384731e57f3177e0c39f2 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Wed, 14 May 2014 00:43:21 +0800
Subject: [PATCH 628/641] UniCore64: fix nocache type

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/mm/mmu.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/unicore64/mm/mmu.c b/arch/unicore64/mm/mmu.c
index 972af89..44d2c88 100644
--- a/arch/unicore64/mm/mmu.c
+++ b/arch/unicore64/mm/mmu.c
@@ -75,7 +75,8 @@ void __init paging_init(void)
 	}
 
 	/* Set first 4M physical memory as noncached */
-	uc64_create_direct_mapping(UC64_PM_NONCACHED_START, SZ_4M, UC64_PM_PGTABLE_PUD_DM00, 0);
+	uc64_create_direct_mapping(UC64_PM_NONCACHED_START, SZ_4M,
+			UC64_PM_PGTABLE_PUD_DM00, UC64_PMD_TYPE_NOCACHE);
 
 	uc64_create_io_direct_mapping();
 
-- 
1.7.9.5

