From e8e6a6f6817f327211d69ad3b43f85440750a77b Mon Sep 17 00:00:00 2001
From: Zhan Yuefeng <zhanyuefeng@mprc.pku.edu.cn>
Date: Thu, 15 May 2014 04:46:57 +0800
Subject: [PATCH 634/641] UniCore64: Realize update_mmu_cache()

Signed-off-by: Zhan Yuefeng <zhanyuefeng@mprc.pku.edu.cn>
---
 arch/unicore64/mm/mmu.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/unicore64/mm/mmu.c b/arch/unicore64/mm/mmu.c
index f2aa87d..24f9238 100644
--- a/arch/unicore64/mm/mmu.c
+++ b/arch/unicore64/mm/mmu.c
@@ -106,5 +106,5 @@ void __init paging_init(void)
 void update_mmu_cache(struct vm_area_struct *vma, unsigned long addr,
 	pte_t *ptep)
 {
-	/* FIXME */
+	__invalid_tlb();
 }
-- 
1.7.9.5

