From e12df96d68ff33cc7590f9762ae02455a5978d93 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Thu, 29 May 2014 08:21:43 +0800
Subject: [PATCH 637/641] UniCore64: Add flush to update_mmu_cache

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/mm/mmu.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/arch/unicore64/mm/mmu.c b/arch/unicore64/mm/mmu.c
index 24f9238..ce48b05 100644
--- a/arch/unicore64/mm/mmu.c
+++ b/arch/unicore64/mm/mmu.c
@@ -1,6 +1,8 @@
 #include <linux/mm.h>
 #include <linux/memblock.h>
 #include <linux/sched.h>
+#include <asm/cacheflush.h>
+#include <asm/tlbflush.h>
 
 #include <asm/setup_arch.h>
 #include <arch/asm-mmuops.h>
@@ -106,5 +108,6 @@ void __init paging_init(void)
 void update_mmu_cache(struct vm_area_struct *vma, unsigned long addr,
 	pte_t *ptep)
 {
-	__invalid_tlb();
+	flush_tlb_mm(vma->vm_mm);
+	flush_cache_all();
 }
-- 
1.7.9.5

