From 854db1ab7f281018c85e7589aa5d0fb5db86314d Mon Sep 17 00:00:00 2001
From: Qin Rui <qinrui12@mprc.pku.edu.cn>
Date: Thu, 29 May 2014 08:20:21 +0800
Subject: [PATCH 638/641] UniCore64: Add __flush_dcache in signal frame setup

Signed-off-by: Qin Rui <qinrui12@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/signal.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/unicore64/kernel/signal.c b/arch/unicore64/kernel/signal.c
index e02fdf4..23582fcb6 100644
--- a/arch/unicore64/kernel/signal.c
+++ b/arch/unicore64/kernel/signal.c
@@ -3,6 +3,7 @@
 
 #include <asm/ucontext.h>
 #include <asm/uaccess.h>
+#include <arch/asm-mmuops.h>
 
 SYSCALL_DEFINE3(sigaltstack, const stack_t __user *, uss,
 		stack_t __user *, uoss,	struct pt_regs *, regs)
@@ -76,6 +77,8 @@ static int setup_rt_frame(int sig, struct k_sigaction *ka, siginfo_t *info,
 	/*Save jepriv code in frame->retcode*/
 	err |= __put_user(0xf000008b, &(frame->retcode));
 
+	__flush_dcache();
+
 	if (err)
 		return -EFAULT;
 
-- 
1.7.9.5

