From 293bdb1b5dc90eba3259dc16d4c55d76f5eaf458 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Wed, 11 Jun 2014 11:20:00 +0800
Subject: [PATCH 640/641] UniCore64: Implement smp_send_stop

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/smp.c |   10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/arch/unicore64/kernel/smp.c b/arch/unicore64/kernel/smp.c
index b238958..2d7c296 100644
--- a/arch/unicore64/kernel/smp.c
+++ b/arch/unicore64/kernel/smp.c
@@ -126,6 +126,9 @@ void ipi_handler(struct pt_regs *regs)
 				irq_exit();
 				break;
 
+			case IPI_CPU_STOP:
+				__halt();
+
 			default:
 				printk(KERN_CRIT "Unknown IPI on CPU %d: %lu\n",
 					cpu, which);
@@ -252,8 +255,11 @@ void __init secondary_start_kernel(void)
 
 void smp_send_stop(void)
 {
-	/* FIXME */
-	BUG();
+	cpumask_t to_whom;
+	cpumask_copy(&to_whom, cpu_possible_mask);
+	cpumask_clear_cpu(smp_processor_id(), &to_whom);
+
+	send_ipi_message(&to_whom, IPI_CPU_STOP);
 }
 
 static void ipi_flush_tlb_mm(void *x)
-- 
1.7.9.5

