From 206006e9d63b5d4934523332f79025c7e697b812 Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Thu, 12 Jun 2014 09:27:07 +0800
Subject: [PATCH 641/641] UniCore64: Cache coherence to ensure tlb consistence

Signed-off-by: Guan Xuetao <gxt@mprc.pku.edu.cn>
---
 arch/unicore64/mm/mmu.c |    1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/unicore64/mm/mmu.c b/arch/unicore64/mm/mmu.c
index b629d2e..bf5f709 100644
--- a/arch/unicore64/mm/mmu.c
+++ b/arch/unicore64/mm/mmu.c
@@ -110,5 +110,4 @@ void update_mmu_cache(struct vm_area_struct *vma, unsigned long addr,
 {
 	/* It's too slow to use flush_tlb_mm(vma->vm_mm) */
 	__invalid_tlb();
-	flush_cache_all();
 }
-- 
1.7.9.5

