From dc6df5b7731d54e0435f3c25022406ce3a5fd2ea Mon Sep 17 00:00:00 2001
From: Zhan Yuefeng <zhanyuefeng@mprc.pku.edu.cn>
Date: Thu, 15 May 2014 02:58:42 +0800
Subject: [PATCH 632/641] UniCore64: Use new noncached region in dma

---
 lib/swiotlb.c |    8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/lib/swiotlb.c b/lib/swiotlb.c
index 6f675e4..5fdd85b 100644
--- a/lib/swiotlb.c
+++ b/lib/swiotlb.c
@@ -160,7 +160,7 @@ void __init swiotlb_init_with_tbl(char *tlb, unsigned long nslabs, int verbose)
 	 * Get the overflow emergency buffer
 	 */
 #ifdef CONFIG_UNICORE64
-	io_tlb_overflow_buffer = (void *)(UC64_VM_NONCACHED_START + SZ_4M - PAGE_ALIGN(io_tlb_overflow));
+	io_tlb_overflow_buffer = (void *)UC64_VM_DMAOVERFLOW_START;
 #else
 	io_tlb_overflow_buffer = alloc_bootmem_low_pages(PAGE_ALIGN(io_tlb_overflow));
 	if (!io_tlb_overflow_buffer)
@@ -190,7 +190,7 @@ swiotlb_init_with_default_size(size_t default_size, int verbose)
 	 * Get IO TLB memory from the low pages
 	 */
 #ifdef CONFIG_UNICORE64
-	io_tlb_start = (char *)UC64_VM_NONCACHED_START;
+	io_tlb_start = (char *)UC64_VM_DMA_START;
 #else
 	io_tlb_start = alloc_bootmem_low_pages(PAGE_ALIGN(bytes));
 	if (!io_tlb_start)
@@ -203,11 +203,7 @@ swiotlb_init_with_default_size(size_t default_size, int verbose)
 void __init
 swiotlb_init(int verbose)
 {
-#ifdef CONFIG_UNICORE64
-	swiotlb_init_with_default_size(SZ_4M - io_tlb_overflow, verbose);
-#else
 	swiotlb_init_with_default_size(64 * (1<<20), verbose);	/* default to 64MB */
-#endif
 }
 
 /*
-- 
1.7.9.5

